<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.37">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Relative Abundances – Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./03.2.AlphaDiversity.html" rel="next">
<link href="./03.InteriorMangroves.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-29e2c20b02301cfff04dc8050bf30c7e.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-43df0d6f53b733b42d37fa7808e17715.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-42e8decf3429976a47c1c54a1701cbb4.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-da771dd4e3759026f460526871f487d2.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<meta property="og:title" content="Relative Abundances – Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves">
<meta property="og:image" content="03.1.RelativeAbundances_files/figure-html/unnamed-chunk-7-1.png">
<meta property="og:site_name" content="Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves">
<meta name="twitter:title" content="Relative Abundances – Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves">
<meta name="twitter:image" content="03.1.RelativeAbundances_files/figure-html/unnamed-chunk-7-1.png">
<meta name="twitter:creator" content="@HOaxacaDiana">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./03.InteriorMangroves.html">Interior mangroves explore</a></li><li class="breadcrumb-item"><a href="./03.1.RelativeAbundances.html">What makes inland mangroves different and unique?</a></li><li class="breadcrumb-item"><a href="./03.1.RelativeAbundances.html">Relative Abundances</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./images/landalab.jpg" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      <div class="sidebar-tools-main">
    <a href="https://github.com/DianaOaxaca/InteriorMangroves" title="GitHub organization" class="quarto-navigation-tool px-1" aria-label="GitHub organization"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./00.Presentation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Presentation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01.Preprocessing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preprocessing</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./A_InteriorCoastalCompare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interior and coastal mangroves compare</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">How similar are coastal and inland mangroves?</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.SurfaceComparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Prepare</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth3 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.1.AlphaDiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alpha diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.2.BetaDiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Beta diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.3.Coremicrobiome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Core microbiome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.4.DifferentialAbundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Differential Abundance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.5.Fig2.InteriorCoastalCompare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Figure Interior and Coastal mangroves compare</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./03.InteriorMangroves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interior mangroves explore</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">What makes inland mangroves different and unique?</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.1.RelativeAbundances.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Relative Abundances</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.2.AlphaDiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alpha Diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.3.BetaDiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Beta diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.4.Coremicrobiome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Core Microbiome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.5.DifferentialAbundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Differential Abundance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.6.MetabolicInference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metabolic Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.7.Fig3.InteriorMangroves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Fig. 3 Interior Mangroves</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#load-libraries" id="toc-load-libraries" class="nav-link active" data-scroll-target="#load-libraries">01. Load libraries</a></li>
  <li>
<a href="#prepare-data" id="toc-prepare-data" class="nav-link" data-scroll-target="#prepare-data">02. Prepare data</a>
  <ul class="collapse">
<li><a href="#create-phyloseq-object" id="toc-create-phyloseq-object" class="nav-link" data-scroll-target="#create-phyloseq-object"><strong>02.1 Create Phyloseq Object</strong></a></li>
  <li><a href="#check-data" id="toc-check-data" class="nav-link" data-scroll-target="#check-data">02.2 Check data</a></li>
  <li><a href="#rename-mangrove-system" id="toc-rename-mangrove-system" class="nav-link" data-scroll-target="#rename-mangrove-system">02.3 Rename mangrove system</a></li>
  <li><a href="#color-pallets" id="toc-color-pallets" class="nav-link" data-scroll-target="#color-pallets">02.4. Color pallets</a></li>
  <li><a href="#check-samples" id="toc-check-samples" class="nav-link" data-scroll-target="#check-samples">02.5 Check samples</a></li>
  <li><a href="#get-full-core" id="toc-get-full-core" class="nav-link" data-scroll-target="#get-full-core">02.6 Get full core</a></li>
  <li><a href="#save-object-in-rds-to-next-analysis" id="toc-save-object-in-rds-to-next-analysis" class="nav-link" data-scroll-target="#save-object-in-rds-to-next-analysis">02.7 Save object in RDS to next analysis</a></li>
  </ul>
</li>
  <li>
<a href="#relative-abundance-phyloseq-objects" id="toc-relative-abundance-phyloseq-objects" class="nav-link" data-scroll-target="#relative-abundance-phyloseq-objects">03. Relative abundance phyloseq objects</a>
  <ul class="collapse">
<li><a href="#create-ampvis-objects" id="toc-create-ampvis-objects" class="nav-link" data-scroll-target="#create-ampvis-objects">03.1 Create ampvis objects</a></li>
  <li><a href="#kingdom" id="toc-kingdom" class="nav-link" data-scroll-target="#kingdom">03.2 Kingdom</a></li>
  <li><a href="#phylum-full" id="toc-phylum-full" class="nav-link" data-scroll-target="#phylum-full">03.3 Phylum full</a></li>
  <li><a href="#phylum-core" id="toc-phylum-core" class="nav-link" data-scroll-target="#phylum-core">03.4 Phylum Core</a></li>
  <li><a href="#combine-plots" id="toc-combine-plots" class="nav-link" data-scroll-target="#combine-plots">03.5 Combine plots</a></li>
  </ul>
</li>
  <li><a href="#dominants" id="toc-dominants" class="nav-link" data-scroll-target="#dominants">Dominants</a></li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./03.InteriorMangroves.html">Interior mangroves explore</a></li><li class="breadcrumb-item"><a href="./03.1.RelativeAbundances.html">What makes inland mangroves different and unique?</a></li><li class="breadcrumb-item"><a href="./03.1.RelativeAbundances.html">Relative Abundances</a></li></ol></nav><div class="quarto-title">
<h1 class="title">Relative Abundances</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="load-libraries" class="level2"><h2 class="anchored" data-anchor-id="load-libraries">01. Load libraries</h2>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">qiime2R</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://microbiome.github.io/microbiome">microbiome</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/">tibble</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://kasperskytte.github.io/ampvis2">ampvis2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">microbiomeutilities</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org">tidyr</a></span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="prepare-data" class="level2"><h2 class="anchored" data-anchor-id="prepare-data">02. Prepare data</h2>
<section id="create-phyloseq-object" class="level3"><h3 class="anchored" data-anchor-id="create-phyloseq-object"><strong>02.1 Create Phyloseq Object</strong></h3>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qiime2R/man/qza_to_phyloseq.html">qza_to_phyloseq</a></span><span class="op">(</span></span>
<span>  features <span class="op">=</span> <span class="st">"data/ASV_table_filter_freq218_emc.qza"</span>,</span>
<span>  tree <span class="op">=</span> <span class="st">"data/rooted-tree-iqtree.qza"</span>,</span>
<span>  taxonomy <span class="op">=</span> <span class="st">"data/taxonomy.qza"</span>,</span>
<span>  metadata <span class="op">=</span> <span class="st">"data/metadata2.tsv"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># check</span></span>
<span><span class="va">ps</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7097 taxa and 50 samples ]
sample_data() Sample Data:       [ 50 samples by 29 sample variables ]
tax_table()   Taxonomy Table:    [ 7097 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 7097 tips and 7096 internal nodes ]</code></pre>
</div>
</div>
</section><section id="check-data" class="level3"><h3 class="anchored" data-anchor-id="check-data">02.2 Check data</h3>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#library(microbiome)</span></span>
<span><span class="fu">microbiome</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/summarize_phyloseq.html">summarize_phyloseq</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Compositional = NO2</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>1] Min. number of reads = 664132] Max. number of reads = 6748463] Total number of reads = 90064984] Average number of reads = 180129.965] Median number of reads = 171486.57] Sparsity = 0.5668845991263916] Any OTU sum to 1 or less? NO8] Number of singletons = 09] Percent of OTUs that are singletons 
        (i.e. exactly one read detected across all samples)010] Number of sample variables are: 29SampleIDDepthDepthsSiteSample_typeDateLocationTemp...C..intSalnidad..ppm.intSalinidad..ppt.intpH.intpHsupRedox.medido.en.campo1Redox.mV..Eh.intTemperatura...C.supSalnidad..ppm.supSalinidad..ppt.supRedox.medido.en.campoRedox.mV..Eh.supS.2..mg.l..campoml.de.muestra1Factor.de.dilucion1S.2..mg.l.ID_MSO.....mg.l..campoml.de.muestra2Factor.de.dilucion2SO.....mg.l.2</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>[[1]]
[1] "1] Min. number of reads = 66413"

[[2]]
[1] "2] Max. number of reads = 674846"

[[3]]
[1] "3] Total number of reads = 9006498"

[[4]]
[1] "4] Average number of reads = 180129.96"

[[5]]
[1] "5] Median number of reads = 171486.5"

[[6]]
[1] "7] Sparsity = 0.566884599126391"

[[7]]
[1] "6] Any OTU sum to 1 or less? NO"

[[8]]
[1] "8] Number of singletons = 0"

[[9]]
[1] "9] Percent of OTUs that are singletons \n        (i.e. exactly one read detected across all samples)0"

[[10]]
[1] "10] Number of sample variables are: 29"

[[11]]
 [1] "Sample"                 "ID"                     "Depth"                 
 [4] "Depths"                 "Site"                   "Sample_type"           
 [7] "Date"                   "Location"               "Temp...C..int"         
[10] "Salnidad..ppm.int"      "Salinidad..ppt.int"     "pH.int"                
[13] "pHsup"                  "Redox.medido.en.campo1" "Redox.mV..Eh.int"      
[16] "Temperatura...C.sup"    "Salnidad..ppm.sup"      "Salinidad..ppt.sup"    
[19] "Redox.medido.en.campo"  "Redox.mV..Eh.sup"       "S.2..mg.l..campo"      
[22] "ml.de.muestra1"         "Factor.de.dilucion1"    "S.2..mg.l."            
[25] "ID_M"                   "SO.....mg.l..campo"     "ml.de.muestra2"        
[28] "Factor.de.dilucion2"    "SO.....mg.l."          </code></pre>
</div>
</div>
</section><section id="rename-mangrove-system" class="level3"><h3 class="anchored" data-anchor-id="rename-mangrove-system">02.3 Rename mangrove system</h3>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Extract metadata</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu">as</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>, <span class="st">"data.frame"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Rename specific levels in the Mangrove system column</span></span>
<span><span class="co">#library(dplyr)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="va">metadata</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span>`Location` <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span></span>
<span>    <span class="va">Location</span> <span class="op">==</span> <span class="st">"El Cacahuate"</span> <span class="op">~</span> <span class="st">"Fossil Lagoon"</span>,</span>
<span>    <span class="va">Location</span> <span class="op">==</span> <span class="st">"La Piedad"</span> <span class="op">~</span> <span class="st">"Reforma Waterfalls"</span>,</span>
<span>    <span class="va">Location</span> <span class="op">==</span> <span class="st">"Dique Miguelito"</span> <span class="op">~</span> <span class="st">"Miguelito Dike"</span>,</span>
<span>    <span class="cn">TRUE</span> <span class="op">~</span> <span class="va">Location</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># update sample_data in phyloseq object</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">metadata</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="color-pallets" class="level3"><h3 class="anchored" data-anchor-id="color-pallets">02.4. Color pallets</h3>
<div class="cell">
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">depth_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0-15"</span> <span class="op">=</span> <span class="st">"#f5e5c4"</span>,</span>
<span>                 <span class="st">"16-30"</span> <span class="op">=</span> <span class="st">"#baa179"</span>,</span>
<span>                 <span class="st">"31-45"</span> <span class="op">=</span> <span class="st">"#a67451"</span>,</span>
<span>                 <span class="st">"50-75"</span> <span class="op">=</span> <span class="st">"#80673b"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">loc_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Fossil Lagoon"</span><span class="op">=</span> <span class="st">"#A3D9A8"</span>,</span>
<span>                <span class="st">"Reforma Waterfalls"</span> <span class="op">=</span> <span class="st">"#44C17B"</span>,</span>
<span>                <span class="st">"Miguelito Dike"</span> <span class="op">=</span> <span class="st">"#2E8B57"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="check-samples" class="level3"><h3 class="anchored" data-anchor-id="check-samples">02.5 Check samples</h3>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Samples Plot</span></span>
<span></span>
<span><span class="co"># Reorder samples</span></span>
<span><span class="va">metadata</span><span class="op">$</span><span class="va">Location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span></span>
<span>  <span class="va">metadata</span><span class="op">$</span><span class="va">Location</span>,</span>
<span>  levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Fossil Lagoon"</span>, <span class="st">"Reforma Waterfalls"</span>, </span>
<span>             <span class="st">"Miguelito Dike"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#plot</span></span>
<span><span class="va">samples_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Location</span>, y <span class="op">=</span> <span class="va">..count..</span>,</span>
<span>fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html">geom_bar</a></span><span class="op">(</span>position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html">facet_wrap</a></span><span class="op">(</span><span class="op">~</span> <span class="va">Location</span>, scales <span class="op">=</span> <span class="st">"free_x"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Samples of the analysis"</span>, y <span class="op">=</span> <span class="st">"Number of Samples"</span>,</span>
<span>       x <span class="op">=</span> <span class="st">"Mangrove system"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">loc_colors</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show plot</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">samples_plot</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: The dot-dot notation (`..count..`) was deprecated in ggplot2 3.4.0.
ℹ Please use `after_stat(count)` instead.</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.1.RelativeAbundances_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="get-full-core" class="level3"><h3 class="anchored" data-anchor-id="get-full-core">02.6 Get full core</h3>
<div class="cell">
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Get full core with microbiome</span></span>
<span><span class="va">full_ps_core_counts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiome/man/core.html">core</a></span><span class="op">(</span><span class="va">ps</span>, <span class="fl">0</span>, <span class="fl">0.5</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Check</span></span>
<span><span class="va">ps</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 7097 taxa and 50 samples ]
sample_data() Sample Data:       [ 50 samples by 29 sample variables ]
tax_table()   Taxonomy Table:    [ 7097 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 7097 tips and 7096 internal nodes ]</code></pre>
</div>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">full_ps_core_counts</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>phyloseq-class experiment-level object
otu_table()   OTU Table:         [ 2048 taxa and 50 samples ]
sample_data() Sample Data:       [ 50 samples by 29 sample variables ]
tax_table()   Taxonomy Table:    [ 2048 taxa by 7 taxonomic ranks ]
phy_tree()    Phylogenetic Tree: [ 2048 tips and 2047 internal nodes ]</code></pre>
</div>
</div>
</section><section id="save-object-in-rds-to-next-analysis" class="level3"><h3 class="anchored" data-anchor-id="save-object-in-rds-to-next-analysis">02.7 Save object in RDS to next analysis</h3>
<div class="cell">
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">ps</span>, <span class="st">"rds/interior_mangroves/phyloseq.rds"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section><section id="relative-abundance-phyloseq-objects" class="level2"><h2 class="anchored" data-anchor-id="relative-abundance-phyloseq-objects">03. Relative abundance phyloseq objects</h2>
<section id="create-ampvis-objects" class="level3"><h3 class="anchored" data-anchor-id="create-ampvis-objects">03.1 Create ampvis objects</h3>
<p>Create full, core and Site ampvis objects</p>
<div class="cell">
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># library(tibble)</span></span>
<span><span class="co"># library(ampvis2)</span></span>
<span><span class="co"># Ampvis object function</span></span>
<span><span class="va">create_ampvis_object</span> <span class="op">&lt;-</span> <span class="kw">function</span><span class="op">(</span><span class="va">pseq</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="co"># create and extract otu table</span></span>
<span>  <span class="va">otu_table_ampvis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>OTU <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">pseq</span><span class="op">)</span><span class="op">@</span><span class="va">.Data</span><span class="op">)</span>,</span>
<span>                                 <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">pseq</span><span class="op">)</span><span class="op">@</span><span class="va">.Data</span>,</span>
<span>                                 <span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/tax_table-methods.html">tax_table</a></span><span class="op">(</span><span class="va">pseq</span><span class="op">)</span><span class="op">@</span><span class="va">.Data</span>,</span>
<span>                                 check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># Metadata</span></span>
<span>  <span class="va">meta_data_ampvis</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span><span class="fu">phyloseq</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">pseq</span><span class="op">)</span>,</span>
<span>                                 check.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># change index by SampleID</span></span>
<span>  <span class="va">meta_data_ampvis</span> <span class="op">&lt;-</span> <span class="va">meta_data_ampvis</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"SampleID"</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># ampvis object</span></span>
<span>  <span class="va">av2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://kasperskytte.github.io/ampvis2/reference/amp_load.html">amp_load</a></span><span class="op">(</span><span class="va">otu_table_ampvis</span>, <span class="va">meta_data_ampvis</span><span class="op">)</span></span>
<span>  </span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html">return</a></span><span class="op">(</span><span class="va">av2</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># phyloseq objects</span></span>
<span><span class="va">phyloseq_objects</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>ps<span class="op">=</span><span class="va">ps</span>, full_ps_core_counts <span class="op">=</span> <span class="va">full_ps_core_counts</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Apply function</span></span>
<span><span class="va">ampvis_objects</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">phyloseq_objects</span>, <span class="va">create_ampvis_object</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ampvis_objects</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>$ps
ampvis2 object with 3 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads Median#Reads 
          50         7097      9006498        66413       674846     171486.5 
   Avg#Reads 
   180129.96 

Assigned taxonomy:
     Kingdom       Phylum        Class        Order       Family        Genus 
  7097(100%) 7077(99.72%) 7043(99.24%) 6811(95.97%) 6795(95.74%) 6672(94.01%) 
     Species 
4818(67.89%) 

Metadata variables: 30 
 SampleID, Sample, ID, Depth, Depths, Site, Sample_type, Date, Location, Temp...C..int, Salnidad..ppm.int, Salinidad..ppt.int, pH.int, pHsup, Redox.medido.en.campo1, Redox.mV..Eh.int, Temperatura...C.sup, Salnidad..ppm.sup, Salinidad..ppt.sup, Redox.medido.en.campo, Redox.mV..Eh.sup, S.2..mg.l..campo, ml.de.muestra1, Factor.de.dilucion1, S.2..mg.l., ID_M, SO.....mg.l..campo, ml.de.muestra2, Factor.de.dilucion2, SO.....mg.l.

$full_ps_core_counts
ampvis2 object with 3 elements. 
Summary of OTU table:
     Samples         OTUs  Total#Reads    Min#Reads    Max#Reads Median#Reads 
          50         2048      6092185        45680       451490       114651 
   Avg#Reads 
    121843.7 

Assigned taxonomy:
     Kingdom       Phylum        Class        Order       Family        Genus 
  2048(100%) 2043(99.76%) 2031(99.17%)  1960(95.7%) 1957(95.56%) 1913(93.41%) 
     Species 
1336(65.23%) 

Metadata variables: 30 
 SampleID, Sample, ID, Depth, Depths, Site, Sample_type, Date, Location, Temp...C..int, Salnidad..ppm.int, Salinidad..ppt.int, pH.int, pHsup, Redox.medido.en.campo1, Redox.mV..Eh.int, Temperatura...C.sup, Salnidad..ppm.sup, Salinidad..ppt.sup, Redox.medido.en.campo, Redox.mV..Eh.sup, S.2..mg.l..campo, ml.de.muestra1, Factor.de.dilucion1, S.2..mg.l., ID_M, SO.....mg.l..campo, ml.de.muestra2, Factor.de.dilucion2, SO.....mg.l.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#subset amp obj</span></span>
<span></span>
<span><span class="va">av2_full</span> <span class="op">&lt;-</span> <span class="va">ampvis_objects</span><span class="op">$</span><span class="va">ps</span></span>
<span><span class="va">av2_full_core</span> <span class="op">&lt;-</span> <span class="va">ampvis_objects</span><span class="op">$</span><span class="va">full_ps_core_counts</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="kingdom" class="level3"><h3 class="anchored" data-anchor-id="kingdom">03.2 Kingdom</h3>
<div class="cell">
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">av2_full</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">Depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">av2_full</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">Depths</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0-15"</span>, <span class="st">"16-30"</span>, <span class="st">"31-45"</span>, <span class="st">"50-75"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">ampv_heatmap_abundances_king_full</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://kasperskytte.github.io/ampvis2/reference/amp_heatmap.html">amp_heatmap</a></span><span class="op">(</span><span class="va">av2_full</span>,</span>
<span>            group_by <span class="op">=</span> <span class="st">"Depth"</span>,</span>
<span>            facet_by <span class="op">=</span>  <span class="st">"Location"</span>,</span>
<span>            plot_values <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            tax_show <span class="op">=</span> <span class="fl">25</span>,</span>
<span>            tax_aggregate <span class="op">=</span> <span class="st">"Kingdom"</span>,</span>
<span>            plot_colorscale <span class="op">=</span> <span class="st">"log10"</span>,</span>
<span>             plot_values_size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>            color_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"deepskyblue3"</span>, <span class="st">"yellow3"</span> ,<span class="st">"magenta3"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size<span class="op">=</span><span class="fl">9</span>, </span>
<span>                                   hjust <span class="op">=</span> <span class="fl">0.5</span>, vjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">12</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span>,</span>
<span>        legend.position<span class="op">=</span><span class="st">"right"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ampv_heatmap_abundances_king_full</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.1.RelativeAbundances_files/figure-html/unnamed-chunk-13-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"Figures/interior_mangroves/RelAbundances_king.pdf"</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/last_plot.html">last_plot</a></span><span class="op">(</span><span class="op">)</span>,width <span class="op">=</span> <span class="fl">16</span>, height <span class="op">=</span> <span class="fl">4.5</span>, units <span class="op">=</span> <span class="st">"cm"</span> <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Get data</p>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># library(microbiome)</span></span>
<span><span class="co"># library(microbiomeutilities)</span></span>
<span><span class="co"># library(dplyr)</span></span>
<span></span>
<span><span class="va">king_abundances</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiomeutilities/man/taxa_summary.html">taxa_summary</a></span><span class="op">(</span><span class="va">ps</span>,<span class="st">"Kingdom"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Data provided is not compositional 
 will first transform</code></pre>
</div>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">king_abundances</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>         Taxa        Max.Rel.Ab        Mean.Rel.Ab      Median.Rel.Ab
1  d__Archaea 0.132017376126457 0.0231528152808648 0.0141264487609834
2 d__Bacteria 0.999715207898234  0.976847184719135  0.985873551239017
             Std.dev
1 0.0247080819800889
2 0.0247080819800889</code></pre>
</div>
</div>
</section><section id="phylum-full" class="level3"><h3 class="anchored" data-anchor-id="phylum-full">03.3 Phylum full</h3>
<p>plot</p>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">ampv_heatmap_abundances_phylum_loc_full</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://kasperskytte.github.io/ampvis2/reference/amp_heatmap.html">amp_heatmap</a></span><span class="op">(</span><span class="va">av2_full</span>,</span>
<span>            group_by <span class="op">=</span> <span class="st">"Depth"</span>,</span>
<span>            facet_by <span class="op">=</span> <span class="st">"Location"</span>,</span>
<span>            plot_values <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            tax_show <span class="op">=</span> <span class="fl">30</span>,</span>
<span>            showRemainingTaxa <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            tax_aggregate <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>            tax_add <span class="op">=</span> <span class="st">"Kingdom"</span>,</span>
<span>            plot_colorscale <span class="op">=</span> <span class="st">"log10"</span>,</span>
<span>            plot_values_size <span class="op">=</span> <span class="fl">2.6</span>,</span>
<span>            color_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"deepskyblue3"</span>, <span class="st">"yellow3"</span> ,<span class="st">"magenta3"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size<span class="op">=</span><span class="fl">8</span>, </span>
<span>                                   vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">9</span><span class="op">)</span>,</span>
<span>        legend.position<span class="op">=</span><span class="st">"right"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ampv_heatmap_abundances_phylum_loc_full</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.1.RelativeAbundances_files/figure-html/unnamed-chunk-15-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<p>**Get info**</p>
<div class="cell">
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#library(ampvis2)</span></span>
<span><span class="co">#heatmap</span></span>
<span><span class="va">phy_ab_depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://kasperskytte.github.io/ampvis2/reference/amp_heatmap.html">amp_heatmap</a></span><span class="op">(</span><span class="va">av2_full</span>,</span>
<span>            <span class="co">#group_by = "Depth",</span></span>
<span>            facet_by <span class="op">=</span>   <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Depth"</span>,<span class="st">"Site"</span>,<span class="st">"Location"</span><span class="op">)</span>,</span>
<span>            plot_values <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            tax_show <span class="op">=</span> <span class="fl">30</span>,</span>
<span>            textmap <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            tax_aggregate <span class="op">=</span> <span class="st">"Phylum"</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># library(dplyr)</span></span>
<span><span class="co"># library(tidyr)</span></span>
<span><span class="co"># library(tibble)</span></span>
<span></span>
<span><span class="co"># Transform data.frame to long format</span></span>
<span><span class="va">phy_ab_long</span> <span class="op">&lt;-</span> <span class="va">phy_ab_depth</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span><span class="st">"Phylum"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">-</span><span class="va">Phylum</span>, </span>
<span>    names_to <span class="op">=</span> <span class="st">"Sample_Location"</span>, </span>
<span>    values_to <span class="op">=</span> <span class="st">"Abundance"</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/separate.html">separate</a></span><span class="op">(</span><span class="va">Sample_Location</span>, into <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Sample"</span>,<span class="st">"Depth"</span>, <span class="st">"Site"</span>, <span class="st">"Location"</span><span class="op">)</span>, sep <span class="op">=</span> <span class="st">" "</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>    Depth <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">gsub</a></span><span class="op">(</span><span class="st">"-cm"</span>, <span class="st">""</span>, <span class="va">Depth</span><span class="op">)</span>  <span class="co"># Clean depth name</span></span>
<span>  <span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Expected 4 pieces. Additional pieces discarded in 1500 rows [1, 2, 3, 4, 5, 6,
7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, ...].</code></pre>
</div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># mean and sd Depth and Location</span></span>
<span><span class="va">phy_ab_summary_stats</span> <span class="op">&lt;-</span> <span class="va">phy_ab_long</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Phylum</span>, <span class="va">Depth</span>, <span class="va">Location</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span></span>
<span>    mean_abundance <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">Abundance</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    sd_abundance <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">Abundance</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    .groups <span class="op">=</span> <span class="st">"drop"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="co"># show</span></span>
<span><span class="va">phy_ab_summary_stats</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 360 × 5
   Phylum          Depth Location  mean_abundance sd_abundance
   &lt;chr&gt;           &lt;chr&gt; &lt;chr&gt;              &lt;dbl&gt;        &lt;dbl&gt;
 1 Acidobacteriota 0-15  Fossil              6.96        0.808
 2 Acidobacteriota 0-15  Miguelito           5.27        1.75 
 3 Acidobacteriota 0-15  Reforma             5.59        1.35 
 4 Acidobacteriota 16-30 Fossil              7.17        0.870
 5 Acidobacteriota 16-30 Miguelito           6.48        0.845
 6 Acidobacteriota 16-30 Reforma             4.47        0.717
 7 Acidobacteriota 31-45 Fossil              6.65        1.01 
 8 Acidobacteriota 31-45 Miguelito           6.50        1.21 
 9 Acidobacteriota 31-45 Reforma             4.77        1.19 
10 Acidobacteriota 50-75 Fossil              8.87        1.87 
# ℹ 350 more rows</code></pre>
</div>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Format table</span></span>
<span><span class="va">phy_depth_table</span> <span class="op">&lt;-</span> <span class="va">phy_ab_summary_stats</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html">pivot_wider</a></span><span class="op">(</span></span>
<span>    names_from <span class="op">=</span> <span class="va">Location</span>,</span>
<span>    values_from <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="va">mean_abundance</span>, <span class="va">sd_abundance</span><span class="op">)</span>,</span>
<span>    names_sep <span class="op">=</span> <span class="st">"_"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">phy_depth_table</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 120 × 8
   Phylum           Depth mean_abundance_Fossil mean_abundance_Miguelito
   &lt;chr&gt;            &lt;chr&gt;                 &lt;dbl&gt;                    &lt;dbl&gt;
 1 Acidobacteriota  0-15                 6.96                      5.27 
 2 Acidobacteriota  16-30                7.17                      6.48 
 3 Acidobacteriota  31-45                6.65                      6.50 
 4 Acidobacteriota  50-75                8.87                      6.12 
 5 Actinobacteriota 0-15                 3.75                      3.52 
 6 Actinobacteriota 16-30                2.84                      2.31 
 7 Actinobacteriota 31-45                3.73                      2.77 
 8 Actinobacteriota 50-75                3.05                      3.15 
 9 Armatimonadota   0-15                 0.0361                    0.193
10 Armatimonadota   16-30                0.231                     0.407
# ℹ 110 more rows
# ℹ 4 more variables: mean_abundance_Reforma &lt;dbl&gt;, sd_abundance_Fossil &lt;dbl&gt;,
#   sd_abundance_Miguelito &lt;dbl&gt;, sd_abundance_Reforma &lt;dbl&gt;</code></pre>
</div>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">phy_depth_table</span>, <span class="st">"Tables/Phylum_depth_location_stats.tsv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="phylum-core" class="level3"><h3 class="anchored" data-anchor-id="phylum-core">03.4 Phylum Core</h3>
<div class="cell">
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Factor Depth_cm </span></span>
<span><span class="va">av2_full_core</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">Depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">av2_full_core</span><span class="op">$</span><span class="va">metadata</span><span class="op">$</span><span class="va">Depths</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0-15"</span>, <span class="st">"16-30"</span>, <span class="st">"31-45"</span>, <span class="st">"50-75"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Full all samples</span></span>
<span><span class="va">ampv_heatmap_abundances_phylum_core</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://kasperskytte.github.io/ampvis2/reference/amp_heatmap.html">amp_heatmap</a></span><span class="op">(</span><span class="va">av2_full_core</span>,</span>
<span>            group_by <span class="op">=</span> <span class="st">"Depth"</span>,</span>
<span>            facet_by <span class="op">=</span> <span class="st">"Location"</span>,</span>
<span>            plot_values <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            tax_show <span class="op">=</span> <span class="fl">30</span>,</span>
<span>            showRemainingTaxa <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            tax_aggregate <span class="op">=</span> <span class="st">"Phylum"</span>,</span>
<span>            tax_add <span class="op">=</span> <span class="st">"Kingdom"</span>,</span>
<span>            plot_colorscale <span class="op">=</span> <span class="st">"log10"</span>,</span>
<span>            plot_values_size <span class="op">=</span> <span class="fl">2.6</span>,</span>
<span>            color_vector <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"deepskyblue3"</span>, <span class="st">"yellow3"</span> ,<span class="st">"magenta3"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">0</span>, size<span class="op">=</span><span class="fl">8</span>, </span>
<span>                                   vjust <span class="op">=</span> <span class="fl">1</span>, hjust <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span>,</span>
<span>        axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size<span class="op">=</span><span class="fl">9</span><span class="op">)</span>,</span>
<span>        legend.position<span class="op">=</span><span class="st">"right"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ampv_heatmap_abundances_phylum_core</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.1.RelativeAbundances_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="combine-plots" class="level3"><h3 class="anchored" data-anchor-id="combine-plots">03.5 Combine plots</h3>
<div class="cell">
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span></span>
<span><span class="co">#Combine venn plot</span></span>
<span><span class="va">title_abund_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/ggdraw.html">ggdraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/draw_label.html">draw_label</a></span><span class="op">(</span><span class="st">"Relative Abundance"</span>, fontface <span class="op">=</span> <span class="st">'bold'</span>, x <span class="op">=</span> <span class="fl">0.5</span>, hjust <span class="op">=</span> <span class="fl">0.5</span>, size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span>
<span><span class="va">ampv_all_phylum_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">title_abund_plot</span>, <span class="va">ampv_heatmap_abundances_phylum_loc_full</span>, <span class="va">ampv_heatmap_abundances_phylum_core</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">""</span>, <span class="st">"A"</span>, <span class="st">"B"</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">1</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.10</span>, <span class="fl">1.2</span>, <span class="fl">1.2</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">ampv_all_phylum_plots</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.1.RelativeAbundances_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#save</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"Figures/interior_mangroves/Abundances_Phylum_full_and_core.pdf"</span>, <span class="va">ampv_all_phylum_plots</span>, width <span class="op">=</span> <span class="fl">18</span>, height <span class="op">=</span> <span class="fl">30</span>, units <span class="op">=</span> <span class="st">"cm"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section><section id="dominants" class="level2"><h2 class="anchored" data-anchor-id="dominants">Dominants</h2>
<div class="cell">
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#psgen &lt;- aggregate_taxa(p0,"Genus")</span></span>
<span><span class="va">tax_dominant</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/microbiomeutilities/man/dominant_taxa.html">dominant_taxa</a></span><span class="op">(</span><span class="va">ps</span>,level <span class="op">=</span> <span class="st">"Phylum"</span>, group<span class="op">=</span><span class="st">"Location"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">tax_dominant</span><span class="op">$</span><span class="va">dominant_overview</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 5
# Groups:   Location [3]
  Location           dominant_taxa      n rel.freq rel.freq.pct
  &lt;chr&gt;              &lt;chr&gt;          &lt;int&gt;    &lt;dbl&gt; &lt;chr&gt;       
1 Fossil Lagoon      Proteobacteria    13     92.9 93%         
2 Miguelito Dike     Chloroflexi       13     54.2 54%         
3 Miguelito Dike     Proteobacteria     9     37.5 38%         
4 Reforma Waterfalls Proteobacteria     8     66.7 67%         
5 Reforma Waterfalls Chloroflexi        3     25   25%         
6 Miguelito Dike     Firmicutes         2      8.3 8%          </code></pre>
</div>
</div>


</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./03.InteriorMangroves.html" class="pagination-link" aria-label="Interior mangroves explore">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Interior mangroves explore</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./03.2.AlphaDiversity.html" class="pagination-link" aria-label="Alpha Diversity">
        <span class="nav-page-text">Alpha Diversity</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© Copyright 2025, Diana Hernández-Oaxaca</p>
</div>   
    <div class="nav-footer-center">
<p>LandaLab</p>
</div>
    <div class="nav-footer-right">
<p>This page is built with ❤ Science 🦠 🧬 and <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>