<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.37">
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">
<title>Alpha Diversity – Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>

<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./03.3.BetaDiversity.html" rel="next">
<link href="./03.1.RelativeAbundances.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-29e2c20b02301cfff04dc8050bf30c7e.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-43df0d6f53b733b42d37fa7808e17715.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-42e8decf3429976a47c1c54a1701cbb4.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-da771dd4e3759026f460526871f487d2.min.css" rel="prefetch" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark"><script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>
<meta property="og:title" content="Alpha Diversity – Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves">
<meta property="og:description" content="Website of microbial diversity analysis of paper: tal tal tal">
<meta property="og:image" content="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-10-1.png">
<meta property="og:site_name" content="Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves">
<meta name="twitter:title" content="Alpha Diversity – Microbial Diversity from a Fossil Lagoon and comparative Interior vs Coastal Mangroves">
<meta name="twitter:description" content="Website of microbial diversity analysis of paper: tal tal tal">
<meta name="twitter:image" content="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-10-1.png">
<meta name="twitter:creator" content="@HOaxacaDiana">
<meta name="twitter:card" content="summary_large_image">
</head>
<body class="nav-sidebar docked">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top"><nav class="quarto-secondary-nav"><div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./03.InteriorMangroves.html">Interior mangroves explore</a></li><li class="breadcrumb-item"><a href="./03.1.RelativeAbundances.html">What makes inland mangroves different and unique?</a></li><li class="breadcrumb-item"><a href="./03.2.AlphaDiversity.html">Alpha Diversity</a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav></header><!-- content --><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto"><div class="pt-lg-2 mt-2 text-center sidebar-header">
      <a href="./index.html" class="sidebar-logo-link">
      <img src="./images/landalab.jpg" alt="" class="sidebar-logo py-0 d-lg-inline d-none"></a>
      <div class="sidebar-tools-main">
    <a href="https://github.com/DianaOaxaca/InteriorMangroves" title="GitHub organization" class="quarto-navigation-tool px-1" aria-label="GitHub organization"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./00.Presentation.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Presentation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01.Preprocessing.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preprocessing</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./A_InteriorCoastalCompare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interior and coastal mangroves compare</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true">
 <span class="menu-text">How similar are coastal and inland mangroves?</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth2 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.SurfaceComparison.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Data Prepare</span></a>
  </div>
</li>
          <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true">
 <span class="menu-text">Analysis</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth3 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.1.AlphaDiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Alpha diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.2.BetaDiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Beta diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.3.Coremicrobiome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Core microbiome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.4.DifferentialAbundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Differential Abundance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02.5.Fig2.InteriorCoastalCompare.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Figure Interior and Coastal mangroves compare</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
      </ul>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./03.InteriorMangroves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Interior mangroves explore</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-4" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-4" class="collapse list-unstyled sidebar-section depth1 show">
<li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a class="sidebar-item-text sidebar-link text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true">
 <span class="menu-text">What makes inland mangroves different and unique?</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-5" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-5" class="collapse list-unstyled sidebar-section depth2 show">
<li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.1.RelativeAbundances.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Relative Abundances</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.2.AlphaDiversity.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text">Alpha Diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.3.BetaDiversity.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Beta diversity</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.4.Coremicrobiome.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Core Microbiome</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.5.DifferentialAbundance.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Differential Abundance</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.6.MetabolicInference.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Metabolic Inference</span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03.7.Fig3.InteriorMangroves.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Fig. 3 Interior Mangroves</span></a>
  </div>
</li>
      </ul>
</li>
      </ul>
</li>
    </ul>
</div>
</nav><div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active"><h2 id="toc-title">On this page</h2>
   
  <ul>
<li><a href="#load-libraries-and-prepare-data" id="toc-load-libraries-and-prepare-data" class="nav-link active" data-scroll-target="#load-libraries-and-prepare-data">Load libraries and prepare data</a></li>
  <li><a href="#get-hill-numbers" id="toc-get-hill-numbers" class="nav-link" data-scroll-target="#get-hill-numbers">01. Get Hill numbers</a></li>
  <li><a href="#get-hill-means" id="toc-get-hill-means" class="nav-link" data-scroll-target="#get-hill-means">02. Get Hill means</a></li>
  <li><a href="#get-sequencing-effort" id="toc-get-sequencing-effort" class="nav-link" data-scroll-target="#get-sequencing-effort">03. Get sequencing effort</a></li>
  <li><a href="#explore-hill-distribution" id="toc-explore-hill-distribution" class="nav-link" data-scroll-target="#explore-hill-distribution">04. Explore hill distribution</a></li>
  <li><a href="#check-alpha-diversity-correlation-with-sequencing-depth" id="toc-check-alpha-diversity-correlation-with-sequencing-depth" class="nav-link" data-scroll-target="#check-alpha-diversity-correlation-with-sequencing-depth">05. Check alpha diversity correlation with sequencing depth</a></li>
  <li><a href="#check-hill-numbers-normality" id="toc-check-hill-numbers-normality" class="nav-link" data-scroll-target="#check-hill-numbers-normality">06. Check Hill numbers Normality</a></li>
  <li>
<a href="#get-significant-differences" id="toc-get-significant-differences" class="nav-link" data-scroll-target="#get-significant-differences">07. Get significant differences</a>
  <ul class="collapse">
<li>
<a href="#by-depth" id="toc-by-depth" class="nav-link" data-scroll-target="#by-depth">By depth</a>
  <ul class="collapse">
<li><a href="#q0-observed-richness" id="toc-q0-observed-richness" class="nav-link" data-scroll-target="#q0-observed-richness">q0 Observed Richness</a></li>
  <li><a href="#q1-diversity-exp-shannon" id="toc-q1-diversity-exp-shannon" class="nav-link" data-scroll-target="#q1-diversity-exp-shannon">q1 Diversity Exp Shannon</a></li>
  <li><a href="#q2-dominance-inv-simpson" id="toc-q2-dominance-inv-simpson" class="nav-link" data-scroll-target="#q2-dominance-inv-simpson">q2 Dominance Inv Simpson</a></li>
  <li><a href="#qf-faith-diversity" id="toc-qf-faith-diversity" class="nav-link" data-scroll-target="#qf-faith-diversity">qf Faith Diversity</a></li>
  <li><a href="#combine-plots" id="toc-combine-plots" class="nav-link" data-scroll-target="#combine-plots">Combine plots</a></li>
  </ul>
</li>
  <li>
<a href="#by-location" id="toc-by-location" class="nav-link" data-scroll-target="#by-location">By location</a>
  <ul class="collapse">
<li><a href="#q0-observed-richness-1" id="toc-q0-observed-richness-1" class="nav-link" data-scroll-target="#q0-observed-richness-1">q0 Observed Richness</a></li>
  <li><a href="#q1-exp-shannon-diversity" id="toc-q1-exp-shannon-diversity" class="nav-link" data-scroll-target="#q1-exp-shannon-diversity">q1 Exp Shannon diversity</a></li>
  <li><a href="#q2-dominance-inv-simpson-1" id="toc-q2-dominance-inv-simpson-1" class="nav-link" data-scroll-target="#q2-dominance-inv-simpson-1">q2 Dominance Inv Simpson</a></li>
  <li><a href="#qf-faith-diversity-1" id="toc-qf-faith-diversity-1" class="nav-link" data-scroll-target="#qf-faith-diversity-1">qf Faith Diversity</a></li>
  <li><a href="#combine-plots-1" id="toc-combine-plots-1" class="nav-link" data-scroll-target="#combine-plots-1">Combine plots</a></li>
  </ul>
</li>
  </ul>
</li>
  <li><a href="#save-rds-plots" id="toc-save-rds-plots" class="nav-link" data-scroll-target="#save-rds-plots">Save rds plots</a></li>
  </ul></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content"><header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./03.InteriorMangroves.html">Interior mangroves explore</a></li><li class="breadcrumb-item"><a href="./03.1.RelativeAbundances.html">What makes inland mangroves different and unique?</a></li><li class="breadcrumb-item"><a href="./03.2.AlphaDiversity.html">Alpha Diversity</a></li></ol></nav><div class="quarto-title">
<h1 class="title">Alpha Diversity</h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header><section id="load-libraries-and-prepare-data" class="level1"><h1>Load libraries and prepare data</h1>
<div class="cell">
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Load libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://dx.plos.org/10.1371/journal.pone.0061217">phyloseq</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/emmanuelparadis/ape">ape</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">qiime2R</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/anttonalberdi/hilldiv">hilldiv</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rpkgs.datanovia.com/ggpubr/">ggpubr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/cowplot/">cowplot</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/vegandevs/vegan">vegan</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://rvlenth.github.io/emmeans/">emmeans</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://multcomp.R-forge.R-project.org">multcomp</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/lme4/lme4/">lme4</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/runehaubo/lmerTestR">lmerTest</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://easystats.github.io/easystats/">easystats</a></span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#load data</span></span>
<span><span class="va">ps</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">readRDS</a></span><span class="op">(</span><span class="st">"rds/interior_mangroves/phyloseq.rds"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Extract data from phyloseq object</span></span>
<span><span class="co">#library(phyloseq)</span></span>
<span><span class="va">otu_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span>, taxa_are_rows <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html">as</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span>, <span class="st">"data.frame"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Extract tree</span></span>
<span><span class="co">#libary(ape)</span></span>
<span><span class="co">#library(qiime2R)</span></span>
<span><span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/qiime2R/man/read_qza.html">read_qza</a></span><span class="op">(</span><span class="st">"data/rooted-tree-iqtree.qza"</span><span class="op">)</span><span class="op">$</span><span class="va">data</span></span>
<span><span class="va">tree_umMPL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/ape/man/chronoMPL.html">chronoMPL</a></span><span class="op">(</span><span class="va">tree</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># color pallet</span></span>
<span><span class="va">depth_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0-15"</span> <span class="op">=</span> <span class="st">"#f5e5c4"</span>,</span>
<span>                 <span class="st">"16-30"</span> <span class="op">=</span> <span class="st">"#baa179"</span>,</span>
<span>                 <span class="st">"31-45"</span> <span class="op">=</span> <span class="st">"#a67451"</span>,</span>
<span>                 <span class="st">"50-75"</span> <span class="op">=</span> <span class="st">"#80673b"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">loc_colors</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Fossil Lagoon"</span><span class="op">=</span> <span class="st">"#A3D9A8"</span>,</span>
<span>                <span class="st">"Reforma Waterfalls"</span> <span class="op">=</span> <span class="st">"#44C17B"</span>,</span>
<span>                <span class="st">"Miguelito Dike"</span> <span class="op">=</span> <span class="st">"#2E8B57"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="get-hill-numbers" class="level1"><h1>01. Get Hill numbers</h1>
<div class="cell">
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># library(hilldiv)</span></span>
<span><span class="co"># library(tidyverse)</span></span>
<span></span>
<span><span class="co"># Get hill numbers</span></span>
<span><span class="va">q0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hilldiv/man/hill_div.html">hill_div</a></span><span class="op">(</span><span class="va">otu_table</span>, qvalue <span class="op">=</span> <span class="fl">0</span><span class="op">)</span> </span>
<span><span class="va">q1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hilldiv/man/hill_div.html">hill_div</a></span><span class="op">(</span><span class="va">otu_table</span>, qvalue <span class="op">=</span> <span class="fl">1</span><span class="op">)</span> </span>
<span><span class="va">q2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hilldiv/man/hill_div.html">hill_div</a></span><span class="op">(</span><span class="va">otu_table</span>, qvalue <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="va">qf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/hilldiv/man/hill_div.html">hill_div</a></span><span class="op">(</span><span class="va">otu_table</span>, qvalue <span class="op">=</span> <span class="fl">0</span>, tree <span class="op">=</span> <span class="va">tree_umMPL</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Merge metadata with Hill numbers</span></span>
<span><span class="va">q012f_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">q0</span>, <span class="va">q1</span>, <span class="va">q2</span>, <span class="va">qf</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/rownames.html">rownames_to_column</a></span><span class="op">(</span>var <span class="op">=</span> <span class="st">"Sample"</span><span class="op">)</span></span>
<span><span class="va">metadata_with_hill</span> <span class="op">&lt;-</span> <span class="va">q012f_all</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span>          </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html">inner_join</a></span><span class="op">(</span><span class="va">metadata</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Sample"</span><span class="op">=</span><span class="st">"Sample"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#save table</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="st">"Tables/interior_mangroves/Metadata_with_hill.tsv"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">TRUE</span>, col.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># show</span></span>
<span><span class="va">metadata_with_hill</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   Sample   q0   q1  q2  qf  ID Depth Depths   Site Sample_type       Date
1  S1A015 2267  581 173 172 S1A     1   0-15 Site_1    Sediment 28/09/2023
2 S1A1530 3082  925 342 213 S1A     2  16-30 Site_1    Sediment 28/09/2023
3 S1A3050 3234  926 303 228 S1A     3  31-45 Site_1    Sediment 28/09/2023
4 S1A5075 3506  880 228 239 S1A     4  50-75 Site_1    Sediment 28/09/2023
5   S1B07 3011  364  63 225 S1B     1   0-15 Site_1    Sediment 28/09/2023
6 S1B2230 3973 1127 320 266 S1B     2  16-30 Site_1    Sediment 28/09/2023
        Location Temp...C..int Salnidad..ppm.int Salinidad..ppt.int pH.int
1 Miguelito Dike          31.4               818               0.82   6.55
2 Miguelito Dike          31.4               818               0.82   6.55
3 Miguelito Dike          31.4               818               0.82   6.55
4 Miguelito Dike          31.4               818               0.82   6.55
5 Miguelito Dike          30.9               794               0.79   6.69
6 Miguelito Dike          30.9               794               0.79   6.69
  pHsup Redox.medido.en.campo1 Redox.mV..Eh.int Temperatura...C.sup
1  7.34                   -114              121                33.0
2  7.34                   -114              121                33.0
3  7.34                   -114              121                33.0
4  7.34                   -114              121                33.0
5  7.79                   -234                1                33.2
6  7.79                   -234                1                33.2
  Salnidad..ppm.sup Salinidad..ppt.sup Redox.medido.en.campo Redox.mV..Eh.sup
1               735               0.73                  -108              127
2               735               0.73                  -108              127
3               735               0.73                  -108              127
4               735               0.73                  -108              127
5               733               0.73                  -136               99
6               733               0.73                  -136               99
  S.2..mg.l..campo ml.de.muestra1 Factor.de.dilucion1 S.2..mg.l.  ID_M
1             7.73             25                   1       7.73 S1a M
2             7.73             25                   1       7.73 S1a M
3             7.73             25                   1       7.73 S1a M
4             7.73             25                   1       7.73 S1a M
5             7.09             25                   1       7.09 S1b M
6             7.09             25                   1       7.09 S1b M
  SO.....mg.l..campo ml.de.muestra2 Factor.de.dilucion2 SO.....mg.l.
1               53.7              2                  10          537
2               53.7              2                  10          537
3               53.7              2                  10          537
4               53.7              2                  10          537
5               55.0              2                  10          550
6               55.0              2                  10          550</code></pre>
</div>
</div>
</section><section id="get-hill-means" class="level1"><h1>02. Get Hill means</h1>
<div class="cell">
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Reorder q values  </span></span>
<span><span class="va">meta_qs</span> <span class="op">&lt;-</span> <span class="va">metadata_with_hill</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="va">q0</span><span class="op">:</span><span class="va">qf</span>, names_to <span class="op">=</span> <span class="st">"q"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">q</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"q0"</span>, <span class="st">"q1"</span>, <span class="st">"q2"</span>, <span class="st">"qf"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>        qs <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span></span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"q0"</span> <span class="op">~</span> <span class="st">"q0=Observed"</span>,</span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"q1"</span> <span class="op">~</span> <span class="st">"q1=Exp Shannon"</span>,</span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"q2"</span> <span class="op">~</span> <span class="st">"q2=Inv Simpson"</span>,</span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"qf"</span> <span class="op">~</span> <span class="st">"q0f=PD Faith"</span></span>
<span>        <span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Get means of Hill numbers</span></span>
<span><span class="va">means</span> <span class="op">&lt;-</span> <span class="va">meta_qs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="va">Site</span>, <span class="va">Location</span>, <span class="va">qs</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>means <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">value</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>            sd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">value</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">'drop'</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#save table</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">means</span>, <span class="st">"Tables/interior_mangroves/Hill_means_sd.tsv"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">TRUE</span>, col.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span><span class="op">(</span><span class="va">means</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 108 × 6
   Depth Site   Location       qs             means    sd
   &lt;int&gt; &lt;chr&gt;  &lt;chr&gt;          &lt;chr&gt;          &lt;dbl&gt; &lt;dbl&gt;
 1     1 Site_1 Miguelito Dike q0=Observed    2639  526. 
 2     1 Site_1 Miguelito Dike q0f=PD Faith    198.  37.3
 3     1 Site_1 Miguelito Dike q1=Exp Shannon  473. 153. 
 4     1 Site_1 Miguelito Dike q2=Inv Simpson  118.  77.4
 5     1 Site_2 Miguelito Dike q0=Observed    3142. 771. 
 6     1 Site_2 Miguelito Dike q0f=PD Faith    224.  40.4
 7     1 Site_2 Miguelito Dike q1=Exp Shannon  941. 339. 
 8     1 Site_2 Miguelito Dike q2=Inv Simpson  288. 116. 
 9     1 Site_3 Miguelito Dike q0=Observed    2771  781. 
10     1 Site_3 Miguelito Dike q0f=PD Faith    211.  41.9
# ℹ 98 more rows</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Get means of Hill numbers</span></span>
<span><span class="va">hillmeansloc</span> <span class="op">&lt;-</span> <span class="va">meta_qs</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span><span class="op">(</span><span class="va">Depth</span>, <span class="va">Location</span>, <span class="va">qs</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarise</a></span><span class="op">(</span>means <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span><span class="op">(</span><span class="va">value</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>            sd <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/sd.html">sd</a></span><span class="op">(</span><span class="va">value</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>            .groups <span class="op">=</span> <span class="st">'drop'</span><span class="op">)</span></span>
<span><span class="va">hillmeansloc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 48 × 5
   Depth Location           qs             means    sd
   &lt;int&gt; &lt;chr&gt;              &lt;chr&gt;          &lt;dbl&gt; &lt;dbl&gt;
 1     1 Fossil Lagoon      q0=Observed    2174. 319. 
 2     1 Fossil Lagoon      q0f=PD Faith    177.  18.0
 3     1 Fossil Lagoon      q1=Exp Shannon  625. 164. 
 4     1 Fossil Lagoon      q2=Inv Simpson  175.  58.9
 5     1 Miguelito Dike     q0=Observed    2850. 592. 
 6     1 Miguelito Dike     q0f=PD Faith    211.  33.0
 7     1 Miguelito Dike     q1=Exp Shannon  675. 333. 
 8     1 Miguelito Dike     q2=Inv Simpson  185. 116. 
 9     1 Reforma Waterfalls q0=Observed    2653. 517. 
10     1 Reforma Waterfalls q0f=PD Faith    195.  30.7
# ℹ 38 more rows</code></pre>
</div>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#save table</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html">write.table</a></span><span class="op">(</span><span class="va">hillmeansloc</span>, <span class="st">"Tables/interior_mangroves/Hill_means_sd_location.tsv"</span>, quote <span class="op">=</span> <span class="cn">FALSE</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, row.names <span class="op">=</span> <span class="cn">TRUE</span>, col.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="get-sequencing-effort" class="level1"><h1>03. Get sequencing effort</h1>
<div class="cell">
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># faster</span></span>
<span><span class="co">#library(vegan)</span></span>
<span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/methods/as.html">as</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/otu_table-methods.html">otu_table</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">)</span>, <span class="st">"matrix"</span><span class="op">)</span></span>
<span><span class="va">raremax</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">min</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html">rowSums</a></span><span class="op">(</span><span class="va">mat</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># loc colors</span></span>
<span><span class="va">location_factor</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/phyloseq/man/sample_data-methods.html">sample_data</a></span><span class="op">(</span><span class="va">ps</span><span class="op">)</span><span class="op">$</span><span class="va">Location</span></span>
<span><span class="va">sample_colors</span> <span class="op">&lt;-</span> <span class="va">loc_colors</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="va">location_factor</span><span class="op">)</span><span class="op">]</span> </span>
<span></span>
<span><span class="co"># font size</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span><span class="op">(</span>cex <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span> </span>
<span></span>
<span><span class="co"># Rarefaction curves</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/rarefy.html">rarecurve</a></span><span class="op">(</span><span class="va">mat</span>, step <span class="op">=</span> <span class="fl">100</span>, sample <span class="op">=</span> <span class="va">raremax</span>, </span>
<span>                      col <span class="op">=</span> <span class="va">sample_colors</span>, label <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-10-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>   user  system elapsed 
  53.07    2.16   55.53 </code></pre>
</div>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/grDevices/pdf.html">pdf</a></span><span class="op">(</span><span class="st">"Figures/interior_mangroves/Sample_effort.pdf"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/vegan/man/rarefy.html">rarecurve</a></span><span class="op">(</span><span class="va">mat</span>, step <span class="op">=</span> <span class="fl">100</span>, sample <span class="op">=</span> <span class="va">raremax</span>, </span>
<span>                      col <span class="op">=</span> <span class="va">sample_colors</span>, label <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>   user  system elapsed 
   49.8     2.7    52.7 </code></pre>
</div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/grDevices/dev.html">dev.off</a></span><span class="op">(</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>quartz_off_screen 
                2 </code></pre>
</div>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#slow but pretty</span></span>
<span><span class="co">#library(ranacapa)</span></span>
<span><span class="co">#library(ggplot2)</span></span>
<span></span>
<span><span class="co">#effort_sampling&lt;- ggrare(ps3, step = 100, color = "sample", label = "sampleID", se = TRUE)</span></span>
<span><span class="co">#rarecurves_plot &lt;- effort_sampling + facet_wrap(~sample)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="explore-hill-distribution" class="level1"><h1>04. Explore hill distribution</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>Following the calculation of the diversity indices and confirmation of adequate sequencing effort, an investigation was conducted on their distribution along the sampled sites and depths.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># factor to reorder plot</span></span>
<span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">Location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">Location</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"Fossil Lagoon"</span>, <span class="st">"Reforma Waterfalls"</span>, <span class="st">"Miguelito Dike"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">Depth</span>  <span class="op">&lt;-</span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">Depths</span>, </span>
<span>         levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0-15"</span>, <span class="st">"16-30"</span>, <span class="st">"31-45"</span>, <span class="st">"50-75"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co">#plot</span></span>
<span><span class="va">hill_barplot_explore</span> <span class="op">&lt;-</span> <span class="va">metadata_with_hill</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html">pivot_longer</a></span><span class="op">(</span>cols <span class="op">=</span> <span class="va">q0</span><span class="op">:</span><span class="va">qf</span>, names_to <span class="op">=</span> <span class="st">"q"</span>, values_to <span class="op">=</span> <span class="st">"value"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span><span class="op">(</span><span class="va">q</span> <span class="op"><a href="https://rdrr.io/r/base/match.html">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"q0"</span>, <span class="st">"q1"</span>, <span class="st">"q2"</span>, <span class="st">"qf"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html">mutate</a></span><span class="op">(</span></span>
<span>        qs <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/case_when.html">case_when</a></span><span class="op">(</span></span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"q0"</span> <span class="op">~</span> <span class="st">"q0=Observed"</span>,</span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"q1"</span> <span class="op">~</span> <span class="st">"q1=Exp Shannon"</span>,</span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"q2"</span> <span class="op">~</span> <span class="st">"q2=Inv Simpson"</span>,</span>
<span>            <span class="va">q</span> <span class="op">==</span> <span class="st">"qf"</span> <span class="op">~</span> <span class="st">"qf0=PD Faith"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggbarplot.html">ggbarplot</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"Depth"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"value"</span>,</span>
<span>    add <span class="op">=</span> <span class="st">"mean_se"</span>,</span>
<span>    facet.by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"qs"</span>, <span class="st">"Location"</span><span class="op">)</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Depth"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">depth_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">0.8</span>, </span>
<span>              position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html">position_jitter</a></span><span class="op">(</span>width <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html">facet_grid</a></span><span class="op">(</span>rows <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html">vars</a></span><span class="op">(</span><span class="va">qs</span><span class="op">)</span>, cols <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/vars.html">vars</a></span><span class="op">(</span><span class="va">Location</span><span class="op">)</span>, </span>
<span>             scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span></span>
<span>    strip.background <span class="op">=</span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    strip.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face<span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">13</span>,</span>
<span>                margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0</span>, <span class="fl">0.5</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    strip.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>face<span class="op">=</span> <span class="st">"bold"</span>, size <span class="op">=</span> <span class="fl">14</span>,</span>
<span>                angle <span class="op">=</span> <span class="fl">0</span>, margin <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">margin</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">0.5</span>, <span class="fl">0</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span>,</span>
<span>    axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.ticks.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.line <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_line</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span>,</span>
<span>    <span class="co">#panel.border = element_blank(),</span></span>
<span>    panel.spacing.x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/grid/unit.html">unit</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="st">"lines"</span><span class="op">)</span>,</span>
<span>    panel.background <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.title.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>    axis.text.y <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="st">"black"</span>, size <span class="op">=</span> <span class="fl">11</span><span class="op">)</span>,</span>
<span>    legend.position <span class="op">=</span> <span class="st">"bottom"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span></span>
<span>    fill <span class="op">=</span> <span class="st">"Depth"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">""</span>,</span>
<span>    x <span class="op">=</span> <span class="st">""</span>,</span>
<span>    title <span class="op">=</span> <span class="st">"Biodiversity across Sites and Depths"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>   <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.text <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">hill_barplot_explore</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-11-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"Figures/interior_mangroves/Hill_diversity_across_Sites_and_Depths.pdf"</span>, <span class="va">hill_barplot_explore</span>, width <span class="op">=</span> <span class="fl">11</span>, height <span class="op">=</span> <span class="fl">11</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>At this juncture, the available data permits only the observation of the distribution of the phenomenon under study. However, it must be noted that comparisons at this stage may prove misleading due to the presence of other variables that must be considered to accurately assess the observed differences.</p>
</div>
</div>
</section><section id="check-alpha-diversity-correlation-with-sequencing-depth" class="level1"><h1>05. Check alpha diversity correlation with sequencing depth</h1>
<div class="cell">
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Get effective reads</span></span>
<span><span class="va">Effective_reads</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Matrix/man/colSums-methods.html">colSums</a></span><span class="op">(</span><span class="va">otu_table</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">Effective_reads</span> <span class="op">&lt;-</span> <span class="va">Effective_reads</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Alpha diversity depth correlation to order q=0</strong></p>
<div class="cell">
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># library(ggpubr)</span></span>
<span></span>
<span><span class="va">q0_vs_depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>,</span>
<span>                x <span class="op">=</span> <span class="st">"Effective_reads"</span>, y <span class="op">=</span> <span class="st">"q0"</span>, </span>
<span>                xlab<span class="op">=</span> <span class="st">"Sequencing depth (number of reads)"</span>,</span>
<span>                ylab <span class="op">=</span> <span class="st">"q=0"</span>, add <span class="op">=</span> <span class="st">"reg.line"</span>,</span>
<span>   add.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#114e9d"</span>, fill <span class="op">=</span> <span class="st">"lightgray"</span><span class="op">)</span>,</span>
<span>   conf.int <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add confidence interval</span></span>
<span>   cor.coef <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add correlation coefficient. </span></span>
<span>   cor.coeff.args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"pearson"</span>, label.x <span class="op">=</span> <span class="fl">3</span>, label.sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span></span>
<span>   <span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Alpha diversity depth correlation to order q=1</strong></p>
<div class="cell">
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">q1_vs_depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, </span>
<span>                x <span class="op">=</span> <span class="st">"Effective_reads"</span>, y <span class="op">=</span> <span class="st">"q1"</span>, </span>
<span>                xlab<span class="op">=</span> <span class="st">"Sequencing depth (number of reads)"</span>,</span>
<span>                ylab <span class="op">=</span> <span class="st">"q=1"</span>, add <span class="op">=</span> <span class="st">"reg.line"</span>,</span>
<span>   add.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#114e9d"</span>, fill <span class="op">=</span> <span class="st">"lightgray"</span><span class="op">)</span>,</span>
<span>   conf.int <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add confidence interval</span></span>
<span>   cor.coef <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add correlation coefficient</span></span>
<span>   cor.coeff.args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"pearson"</span>, label.x <span class="op">=</span> <span class="fl">3</span>, label.sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>Alpha diversity depth correlation to order q=2</strong></p>
<div class="cell">
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">q2_vs_depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, </span>
<span>                x <span class="op">=</span> <span class="st">"Effective_reads"</span>, y <span class="op">=</span> <span class="st">"q2"</span>, </span>
<span>                xlab<span class="op">=</span> <span class="st">"Sequencing depth (number of reads)"</span>,</span>
<span>                ylab <span class="op">=</span> <span class="st">"q=2"</span>, add <span class="op">=</span> <span class="st">"reg.line"</span>,</span>
<span>   add.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#114e9d"</span>, fill <span class="op">=</span> <span class="st">"lightgray"</span><span class="op">)</span>,</span>
<span>   conf.int <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add confidence interval</span></span>
<span>   cor.coef <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add correlation coefficient</span></span>
<span>   cor.coeff.args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"pearson"</span>, label.x <span class="op">=</span> <span class="fl">3</span>, label.sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qf_vs_depth</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rpkgs.datanovia.com/ggpubr/reference/ggscatter.html">ggscatter</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, </span>
<span>                x <span class="op">=</span> <span class="st">"Effective_reads"</span>, y <span class="op">=</span> <span class="st">"qf"</span>, </span>
<span>                xlab<span class="op">=</span> <span class="st">"Sequencing depth (number of reads)"</span>,</span>
<span>                ylab <span class="op">=</span> <span class="st">"q=phy"</span>, add <span class="op">=</span> <span class="st">"reg.line"</span>,</span>
<span>   add.params <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>color <span class="op">=</span> <span class="st">"#114e9d"</span>, fill <span class="op">=</span> <span class="st">"lightgray"</span><span class="op">)</span>,</span>
<span>   conf.int <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add confidence interval</span></span>
<span>   cor.coef <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Add correlation coefficient</span></span>
<span>   cor.coeff.args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>method <span class="op">=</span> <span class="st">"pearson"</span>, label.x <span class="op">=</span> <span class="fl">3</span>, label.sep <span class="op">=</span> <span class="st">"\n"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.title <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span>, </span>
<span>        legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># library(ggplot2)</span></span>
<span><span class="co"># library(cowplot)</span></span>
<span></span>
<span><span class="co">#Combine plot</span></span>
<span><span class="va">title_corr_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/ggdraw.html">ggdraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://wilkelab.org/cowplot/reference/draw_label.html">draw_label</a></span><span class="op">(</span><span class="st">"Alpha diversity depth correlation"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_plot_q012</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">title_corr_plot</span>,</span>
<span>                            <span class="va">q0_vs_depth</span>, </span>
<span>                            <span class="va">q1_vs_depth</span>, </span>
<span>                            <span class="va">q2_vs_depth</span>,</span>
<span>                            ncol <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                            rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">correlation_plot_q012</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># #save plot</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"Figures/interior_mangroves/alpha_depth_correlation.pdf"</span>, <span class="va">correlation_plot_q012</span>, width <span class="op">=</span> <span class="fl">8</span>, height <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Take note
</div>
</div>
<div class="callout-body-container callout-body">
<p>We found a slight positive correlation between observed richness and sequencing depth, indicating that this variable should be considered when conducting meaningful comparisons of this index.</p>
</div>
</div>
</section><section id="check-hill-numbers-normality" class="level1"><h1>06. Check Hill numbers Normality</h1>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Tip
</div>
</div>
<div class="callout-body-container callout-body">
<p>A methodology that can be used to evaluate the most appropriate test to employ in cases where significant differences between comparisons are identified is to ascertain the normality of the data. In this instance, the normality of the diversity indices was assessed.</p>
</div>
</div>
<p>Shapiro test to Hill numbers</p>
<div class="cell">
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Shapiro test</span></span>
<span><span class="va">shapiro_q0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">q0</span><span class="op">)</span></span>
<span><span class="va">shapiro_q0_pvalue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">shapiro_q0</span><span class="op">$</span><span class="va">p.value</span>, <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">shapiro_q1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">q1</span><span class="op">)</span></span>
<span><span class="va">shapiro_q1_pvalue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">shapiro_q1</span><span class="op">$</span><span class="va">p.value</span>, <span class="fl">5</span><span class="op">)</span></span>
<span><span class="va">shapiro_q2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/shapiro.test.html">shapiro.test</a></span><span class="op">(</span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">q2</span><span class="op">)</span></span>
<span><span class="va">shapiro_q2_pvalue</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html">round</a></span><span class="op">(</span><span class="va">shapiro_q2</span><span class="op">$</span><span class="va">p.value</span>, <span class="fl">5</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Histograms</span></span>
<span><span class="va">histplot_q0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">q0</span>, xlab<span class="op">=</span><span class="st">"q=0"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"lightblue"</span>, bins <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Shapiro, p-value:"</span>, <span class="va">shapiro_q0_pvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"q=0"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Frequency"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">histplot_q1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">q1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"lightblue"</span>, bins <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Shapiro, p-value:"</span>, <span class="va">shapiro_q1_pvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"q=1"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Frequency"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">histplot_q2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">q2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html">geom_histogram</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="st">"lightblue"</span>, bins <span class="op">=</span> <span class="fl">15</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste</a></span><span class="op">(</span><span class="st">"Shapiro, p-value:"</span>, <span class="va">shapiro_q2_pvalue</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"q=2"</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"Frequency"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Combine plot</span></span>
<span><span class="va">title_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/ggdraw.html">ggdraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/draw_label.html">draw_label</a></span><span class="op">(</span><span class="st">"Histogram of Hill diversity"</span><span class="op">)</span> <span class="co">#, fontface = 'bold', x = 0.5, hjust = 0.5)</span></span>
<span><span class="va">histplot_q012</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">title_plot</span>, <span class="va">histplot_q0</span>, <span class="va">histplot_q1</span>, <span class="va">histplot_q2</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">" "</span>,<span class="st">"A"</span>, <span class="st">"B"</span>, <span class="st">"C"</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fl">1</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.15</span>, <span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">histplot_q012</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-18-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#save plot</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"Figures/interior_mangroves/hill_normality_shapiro.png"</span>, <span class="va">histplot_q012</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Saving 7 x 5 in image</code></pre>
</div>
</div>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Caution
</div>
</div>
<div class="callout-body-container callout-body">
<p>Despite the observed normal distribution, sequencing depth, the presence of repeated measures, and imbalances in the project data necessitates model-based comparison. In this instance, the utilization of linear mixed models is a good way to evaluate effects and differences.</p>
</div>
</div>
</section><section id="get-significant-differences" class="level1"><h1>07. Get significant differences</h1>
<div class="callout callout-style-default callout-caution callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Remember
</div>
</div>
<div class="callout-body-container callout-body">
<p>The significance of the observed differences among mangrove systems was determined by employing a linear mixed model, given the imbalanced sample numbers.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Change fossil lagoon reference</span></span>
<span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">Location</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span><span class="op">(</span><span class="va">metadata_with_hill</span><span class="op">$</span><span class="va">Location</span>, ref <span class="op">=</span> <span class="st">"Fossil Lagoon"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="by-depth" class="level2"><h2 class="anchored" data-anchor-id="by-depth">By depth</h2>
<section id="q0-observed-richness" class="level3"><h3 class="anchored" data-anchor-id="q0-observed-richness">q0 Observed Richness</h3>
<div class="cell">
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmm</span></span>
<span><span class="va">q0lmer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">q0</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="va">Effective_reads</span><span class="op">)</span> <span class="op">+</span> <span class="va">Depths</span> <span class="op">+</span> <span class="va">Location</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co"># summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">q0lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: q0 ~ log(Effective_reads) + Depths + Location + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 680

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-2.3093 -0.5309 -0.0431  0.4842  1.8144 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 210009   458     
 Residual             209059   457     
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error       df t value Pr(&gt;|t|)    
(Intercept)                -5467.48    2792.66    37.52   -1.96  0.05772 .  
log(Effective_reads)         643.23     232.78    37.02    2.76  0.00886 ** 
Depths16-30                  444.48     180.16    32.87    2.47  0.01900 *  
Depths31-45                  699.07     181.59    32.94    3.85  0.00052 ***
Depths50-75                  547.05     189.70    33.13    2.88  0.00686 ** 
LocationReforma Waterfalls   752.10     394.68     9.85    1.91  0.08628 .  
LocationMiguelito Dike       396.38     334.63     9.97    1.18  0.26366    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) lg(E_) D16-30 D31-45 D50-75 LctnRW
lg(Effctv_) -0.995                                   
Depths16-30  0.063 -0.095                            
Depths31-45  0.124 -0.157  0.507                     
Depths50-75  0.018 -0.045  0.475  0.474              
LctnRfrmWtr -0.017 -0.044  0.004  0.007 -0.039       
LoctnMgltDk -0.028 -0.044  0.004  0.007 -0.046  0.517</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Permutation test con lmerTest</span></span>
<span><span class="va">perm_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lmerTest/man/ranova.html">rand</a></span><span class="op">(</span><span class="va">q0lmer</span>, nsim <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">perm_test</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
q0 ~ log(Effective_reads) + Depths + Location + (1 | ID)
         npar logLik AIC  LRT Df Pr(&gt;Chisq)   
&lt;none&gt;      9   -340 698                      
(1 | ID)    8   -346 707 10.8  1      0.001 **
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#effects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">q0lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Type III Analysis of Variance Table with Satterthwaite's method
                      Sum Sq Mean Sq NumDF DenDF F value Pr(&gt;F)   
log(Effective_reads) 1596301 1596301     1  37.0    7.64 0.0089 **
Depths               3412939 1137646     3  33.0    5.44 0.0037 **
Location              770456  385228     2   9.8    1.84 0.2094   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># check model reliability</span></span>
<span><span class="fu">report</span><span class="op">(</span><span class="va">q0lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Formula contains log- or sqrt-terms.
  See help("standardize") for how such terms are standardized.
Formula contains log- or sqrt-terms.
  See help("standardize") for how such terms are standardized.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>We fitted a linear mixed model (estimated using REML and nloptwrap optimizer)
to predict q0 with Effective_reads, Depths and Location (formula: q0 ~
log(Effective_reads) + Depths + Location). The model included ID as random
effect (formula: ~1 | ID). The model's total explanatory power is substantial
(conditional R2 = 0.67) and the part related to the fixed effects alone
(marginal R2) is of 0.34. The model's intercept, corresponding to
Effective_reads = 0, Depths = 0-15 and Location = Fossil Lagoon, is at -5467.48
(95% CI [-11107.37, 172.40], t(41) = -1.96, p = 0.057). Within this model:

  - The effect of Effective reads [log] is statistically significant and positive
(beta = 643.23, 95% CI [173.12, 1113.33], t(41) = 2.76, p = 0.009; Std. beta =
0.89, 95% CI [0.23, 1.54])
  - The effect of Depths [16-30] is statistically significant and positive (beta
= 444.48, 95% CI [80.64, 808.32], t(41) = 2.47, p = 0.018; Std. beta = 0.57,
95% CI [0.10, 1.04])
  - The effect of Depths [31-45] is statistically significant and positive (beta
= 699.07, 95% CI [332.34, 1065.79], t(41) = 3.85, p &lt; .001; Std. beta = 0.90,
95% CI [0.43, 1.37])
  - The effect of Depths [50-75] is statistically significant and positive (beta
= 547.05, 95% CI [163.93, 930.16], t(41) = 2.88, p = 0.006; Std. beta = 0.69,
95% CI [0.20, 1.19])
  - The effect of Location [Reforma Waterfalls] is statistically non-significant
and positive (beta = 752.10, 95% CI [-44.97, 1549.17], t(41) = 1.91, p = 0.064;
Std. beta = 0.97, 95% CI [-0.06, 1.99])
  - The effect of Location [Miguelito Dike] is statistically non-significant and
positive (beta = 396.38, 95% CI [-279.41, 1072.18], t(41) = 1.18, p = 0.243;
Std. beta = 0.51, 95% CI [-0.36, 1.38])

Standardized parameters were obtained by fitting the model on a standardized
version of the dataset. 95% Confidence Intervals (CIs) and p-values were
computed using a Wald t-distribution approximation.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># get differences with emmeans</span></span>
<span><span class="va">q0_lmer_means</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">q0lmer</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Depths</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># get significance letters</span></span>
<span><span class="va">cld_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q0_lmer_means</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to data frame</span></span>
<span><span class="va">q0_emmeans_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">cld_results</span><span class="op">)</span></span>
<span><span class="va">q0_emmeans_df</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Depths emmean  SE   df lower.CL upper.CL .group
 0-15     2699 186 22.2     2313     3085  a    
 16-30    3143 184 21.6     2760     3526  ab   
 50-75    3246 196 25.3     2843     3648   b   
 31-45    3398 184 21.5     3015     3780   b   

Results are averaged over the levels of: Location 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 4 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># boxplot with significance</span></span>
<span><span class="va">q0plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Depths</span>, y <span class="op">=</span> <span class="va">q0</span>, fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">q0_emmeans_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">9</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Observed Richness"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span> <span class="op">)</span> <span class="op">+</span> <span class="co">#,</span></span>
<span>       <span class="co">#caption = "letters by lmer emmeans") +</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">depth_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>, axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">q0plot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-24-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="q1-diversity-exp-shannon" class="level3"><h3 class="anchored" data-anchor-id="q1-diversity-exp-shannon">q1 Diversity Exp Shannon</h3>
<div class="cell">
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Fitted linear mixed model</span></span>
<span><span class="va">q1lmer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">q1</span> <span class="op">~</span> <span class="va">Depths</span><span class="op">+</span><span class="va">Location</span>  <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co"># summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">q1lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: q1 ~ Depths + Location + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 627

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-1.6697 -0.6187  0.0827  0.3962  2.4515 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 35546    189     
 Residual             47469    218     
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error    df t value Pr(&gt;|t|)    
(Intercept)                   596.2      121.9  15.3    4.89  0.00019 ***
Depths16-30                   214.1       85.5  34.2    2.51  0.01716 *  
Depths31-45                   165.5       85.5  34.2    1.94  0.06116 .  
Depths50-75                   131.8       90.3  34.5    1.46  0.15340    
LocationReforma Waterfalls     98.3      168.0  10.3    0.59  0.57086    
LocationMiguelito Dike         23.7      142.5  10.5    0.17  0.87137    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) D16-30 D31-45 D50-75 LctnRW
Depths16-30 -0.351                            
Depths31-45 -0.351  0.500                     
Depths50-75 -0.289  0.473  0.473              
LctnRfrmWtr -0.598  0.000  0.000 -0.045       
LoctnMgltDk -0.704  0.000  0.000 -0.053  0.518</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmerTest</span></span>
<span><span class="va">perm_testq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lmerTest/man/ranova.html">rand</a></span><span class="op">(</span><span class="va">q1lmer</span>, nsim <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summary</span></span>
<span><span class="va">perm_testq1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
q1 ~ Depths + Location + (1 | ID)
         npar logLik AIC  LRT Df Pr(&gt;Chisq)   
&lt;none&gt;      8   -313 643                      
(1 | ID)    7   -318 650 8.94  1     0.0028 **
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Efects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">q1lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Type III Analysis of Variance Table with Satterthwaite's method
         Sum Sq Mean Sq NumDF DenDF F value Pr(&gt;F)  
Depths   327747  109249     3  34.3    2.30  0.095 .
Location  17497    8749     2  10.2    0.18  0.834  
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#check_model(q1lmer)</span></span>
<span><span class="fu">report</span><span class="op">(</span><span class="va">q1lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>We fitted a linear mixed model (estimated using REML and nloptwrap optimizer)
to predict q1 with Depths and Location (formula: q1 ~ Depths + Location). The
model included ID as random effect (formula: ~1 | ID). The model's total
explanatory power is substantial (conditional R2 = 0.48) and the part related
to the fixed effects alone (marginal R2) is of 0.09. The model's intercept,
corresponding to Depths = 0-15 and Location = Fossil Lagoon, is at 596.15 (95%
CI [350.16, 842.15], t(42) = 4.89, p &lt; .001). Within this model:

  - The effect of Depths [16-30] is statistically significant and positive (beta
= 214.10, 95% CI [41.64, 386.56], t(42) = 2.51, p = 0.016; Std. beta = 0.76,
95% CI [0.15, 1.38])
  - The effect of Depths [31-45] is statistically non-significant and positive
(beta = 165.46, 95% CI [-7.00, 337.92], t(42) = 1.94, p = 0.060; Std. beta =
0.59, 95% CI [-0.02, 1.20])
  - The effect of Depths [50-75] is statistically non-significant and positive
(beta = 131.76, 95% CI [-50.39, 313.91], t(42) = 1.46, p = 0.152; Std. beta =
0.47, 95% CI [-0.18, 1.12])
  - The effect of Location [Reforma Waterfalls] is statistically non-significant
and positive (beta = 98.34, 95% CI [-240.64, 437.32], t(42) = 0.59, p = 0.561;
Std. beta = 0.35, 95% CI [-0.86, 1.56])
  - The effect of Location [Miguelito Dike] is statistically non-significant and
positive (beta = 23.65, 95% CI [-263.97, 311.27], t(42) = 0.17, p = 0.869; Std.
beta = 0.08, 95% CI [-0.94, 1.11])

Standardized parameters were obtained by fitting the model on a standardized
version of the dataset. 95% Confidence Intervals (CIs) and p-values were
computed using a Wald t-distribution approximation.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Sig dif emmeans</span></span>
<span><span class="va">q1_lmer_means</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">q1lmer</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Depths</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># letters</span></span>
<span><span class="va">cld_resultsq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q1_lmer_means</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert df</span></span>
<span><span class="va">emmeans_dfq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">cld_resultsq1</span><span class="op">)</span></span>
<span><span class="va">emmeans_dfq1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Depths emmean   SE   df lower.CL upper.CL .group
 0-15      637 81.8 24.7      468      805  a    
 50-75     769 87.1 28.6      590      947  a    
 31-45     802 81.8 24.7      634      971  a    
 16-30     851 81.8 24.7      682     1019  a    

Results are averaged over the levels of: Location 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 4 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># plot</span></span>
<span><span class="va">q1plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Depths</span>, y <span class="op">=</span> <span class="va">q1</span>, fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">emmeans_dfq1</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">9</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>,position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Exp Shannon"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span> <span class="op">)</span> <span class="op">+</span> <span class="co">#,</span></span>
<span>       <span class="co">#caption = "letters by lmer emmeans") +</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">depth_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">q1plot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-29-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="q2-dominance-inv-simpson" class="level3"><h3 class="anchored" data-anchor-id="q2-dominance-inv-simpson">q2 Dominance Inv Simpson</h3>
<div class="cell">
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Fitted linear mixed model</span></span>
<span><span class="va">q2lmer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">q2</span> <span class="op">~</span> <span class="va">Depths</span><span class="op">+</span><span class="va">Location</span> <span class="op">+</span><span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co">#summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">q2lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: q2 ~ Depths + Location + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 542

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-1.7063 -0.6064  0.0468  0.5556  2.3862 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 2840     53.3    
 Residual             7832     88.5    
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error     df t value Pr(&gt;|t|)    
(Intercept)                  166.94      41.16  18.56    4.06   0.0007 ***
Depths16-30                   87.60      34.71  34.19    2.52   0.0164 *  
Depths31-45                   31.89      34.71  34.19    0.92   0.3647    
Depths50-75                   35.58      36.61  34.70    0.97   0.3378    
LocationReforma Waterfalls    10.23      53.74  10.38    0.19   0.8527    
LocationMiguelito Dike         1.72      45.70  10.62    0.04   0.9706    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) D16-30 D31-45 D50-75 LctnRW
Depths16-30 -0.422                            
Depths31-45 -0.422  0.500                     
Depths50-75 -0.349  0.474  0.474              
LctnRfrmWtr -0.570  0.000  0.000 -0.056       
LoctnMgltDk -0.671  0.000  0.000 -0.066  0.525</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmerTest</span></span>
<span><span class="va">perm_testq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lmerTest/man/ranova.html">rand</a></span><span class="op">(</span><span class="va">q2lmer</span>, nsim <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summary</span></span>
<span><span class="va">perm_testq2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
q2 ~ Depths + Location + (1 | ID)
         npar logLik AIC LRT Df Pr(&gt;Chisq)  
&lt;none&gt;      8   -271 559                    
(1 | ID)    7   -273 560 3.5  1      0.061 .
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Efects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">q2lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Type III Analysis of Variance Table with Satterthwaite's method
         Sum Sq Mean Sq NumDF DenDF F value Pr(&gt;F)
Depths    51262   17087     3  34.4    2.18   0.11
Location    326     163     2  10.2    0.02   0.98</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#check_model(q2lmer)</span></span>
<span><span class="fu">report</span><span class="op">(</span><span class="va">q2lmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>We fitted a linear mixed model (estimated using REML and nloptwrap optimizer)
to predict q2 with Depths and Location (formula: q2 ~ Depths + Location). The
model included ID as random effect (formula: ~1 | ID). The model's total
explanatory power is substantial (conditional R2 = 0.33) and the part related
to the fixed effects alone (marginal R2) is of 0.09. The model's intercept,
corresponding to Depths = 0-15 and Location = Fossil Lagoon, is at 166.94 (95%
CI [83.88, 250.00], t(42) = 4.06, p &lt; .001). Within this model:

  - The effect of Depths [16-30] is statistically significant and positive (beta
= 87.60, 95% CI [17.55, 157.65], t(42) = 2.52, p = 0.015; Std. beta = 0.86, 95%
CI [0.17, 1.55])
  - The effect of Depths [31-45] is statistically non-significant and positive
(beta = 31.89, 95% CI [-38.16, 101.94], t(42) = 0.92, p = 0.364; Std. beta =
0.31, 95% CI [-0.38, 1.00])
  - The effect of Depths [50-75] is statistically non-significant and positive
(beta = 35.58, 95% CI [-38.29, 109.46], t(42) = 0.97, p = 0.337; Std. beta =
0.35, 95% CI [-0.38, 1.08])
  - The effect of Location [Reforma Waterfalls] is statistically non-significant
and positive (beta = 10.23, 95% CI [-98.22, 118.68], t(42) = 0.19, p = 0.850;
Std. beta = 0.10, 95% CI [-0.97, 1.17])
  - The effect of Location [Miguelito Dike] is statistically non-significant and
positive (beta = 1.72, 95% CI [-90.50, 93.94], t(42) = 0.04, p = 0.970; Std.
beta = 0.02, 95% CI [-0.89, 0.92])

Standardized parameters were obtained by fitting the model on a standardized
version of the dataset. 95% Confidence Intervals (CIs) and p-values were
computed using a Wald t-distribution approximation.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Sig dif emmeans</span></span>
<span><span class="va">q2_lmer_means</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">q2lmer</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Depths</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># letters</span></span>
<span><span class="va">cld_resultsq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q2_lmer_means</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert df</span></span>
<span><span class="va">emmeans_dfq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">cld_resultsq2</span><span class="op">)</span></span>
<span><span class="va">emmeans_dfq2</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Depths emmean   SE   df lower.CL upper.CL .group
 0-15      171 29.2 32.4      112      230  a    
 31-45     203 29.2 32.4      143      262  a    
 50-75     206 31.6 36.2      142      271  a    
 16-30     258 29.2 32.4      199      318  a    

Results are averaged over the levels of: Location 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 4 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># plot</span></span>
<span><span class="va">q2plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Depths</span>, y <span class="op">=</span> <span class="va">q2</span>, fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">emmeans_dfq2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">9</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>,position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"q2 = Inv Simpson"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span> <span class="op">)</span> <span class="op">+</span> <span class="co">#,</span></span>
<span>       <span class="co">#caption = "letters by lmer emmeans") +</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">depth_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">q2plot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-34-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="qf-faith-diversity" class="level3"><h3 class="anchored" data-anchor-id="qf-faith-diversity">qf Faith Diversity</h3>
<div class="cell">
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Fitted linear mixed model</span></span>
<span><span class="va">qflmer</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">qf</span> <span class="op">~</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="va">Effective_reads</span><span class="op">)</span> <span class="op">+</span> <span class="va">Depths</span> <span class="op">+</span> <span class="va">Location</span>  <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co">#summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">qflmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: qf ~ +log(Effective_reads) + Depths + Location + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 428

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-2.6973 -0.3974 -0.0189  0.5074  1.9272 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 620      24.9    
 Residual             589      24.3    
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error      df t value Pr(&gt;|t|)    
(Intercept)                 -152.26     148.43   37.39   -1.03  0.31157    
log(Effective_reads)          28.23      12.37   36.87    2.28  0.02836 *  
Depths16-30                   26.14       9.56   32.85    2.73  0.01003 *  
Depths31-45                   41.53       9.64   32.91    4.31  0.00014 ***
Depths50-75                   32.48      10.07   33.10    3.23  0.00283 ** 
LocationReforma Waterfalls    25.37      21.34    9.83    1.19  0.26247    
LocationMiguelito Dike        11.02      18.09    9.95    0.61  0.55613    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) lg(E_) D16-30 D31-45 D50-75 LctnRW
lg(Effctv_) -0.995                                   
Depths16-30  0.063 -0.095                            
Depths31-45  0.125 -0.157  0.507                     
Depths50-75  0.019 -0.045  0.475  0.474              
LctnRfrmWtr -0.019 -0.043  0.004  0.007 -0.038       
LoctnMgltDk -0.030 -0.043  0.004  0.007 -0.045  0.517</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmerTest</span></span>
<span><span class="va">perm_testqf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lmerTest/man/ranova.html">rand</a></span><span class="op">(</span><span class="va">qflmer</span>, nsim <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Summary</span></span>
<span><span class="va">perm_testqf</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
qf ~ log(Effective_reads) + Depths + Location + (1 | ID)
         npar logLik AIC  LRT Df Pr(&gt;Chisq)    
&lt;none&gt;      9   -214 446                       
(1 | ID)    8   -220 456 11.3  1    0.00078 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode" id="cb72"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Efects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">qflmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Type III Analysis of Variance Table with Satterthwaite's method
                     Sum Sq Mean Sq NumDF DenDF F value Pr(&gt;F)   
log(Effective_reads)   3067    3067     1  36.9    5.21 0.0284 * 
Depths                12031    4010     3  33.0    6.81 0.0011 **
Location                832     416     2   9.8    0.71 0.5169   
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb74"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#check_model(qflmer)</span></span>
<span><span class="fu">report</span><span class="op">(</span><span class="va">qflmer</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Formula contains log- or sqrt-terms.
  See help("standardize") for how such terms are standardized.
Formula contains log- or sqrt-terms.
  See help("standardize") for how such terms are standardized.</code></pre>
</div>
<div class="cell-output cell-output-stdout">
<pre><code>We fitted a linear mixed model (estimated using REML and nloptwrap optimizer)
to predict qf with Effective_reads, Depths and Location (formula: qf ~
+log(Effective_reads) + Depths + Location). The model included ID as random
effect (formula: ~1 | ID). The model's total explanatory power is substantial
(conditional R2 = 0.65) and the part related to the fixed effects alone
(marginal R2) is of 0.28. The model's intercept, corresponding to
Effective_reads = 0, Depths = 0-15 and Location = Fossil Lagoon, is at -152.26
(95% CI [-452.03, 147.50], t(41) = -1.03, p = 0.311). Within this model:

  - The effect of Effective reads [log] is statistically significant and positive
(beta = 28.23, 95% CI [3.25, 53.21], t(41) = 2.28, p = 0.028; Std. beta = 0.76,
95% CI [0.08, 1.44])
  - The effect of Depths [16-30] is statistically significant and positive (beta
= 26.14, 95% CI [6.82, 45.45], t(41) = 2.73, p = 0.009; Std. beta = 0.66, 95%
CI [0.17, 1.14])
  - The effect of Depths [31-45] is statistically significant and positive (beta
= 41.53, 95% CI [22.06, 61.00], t(41) = 4.31, p &lt; .001; Std. beta = 1.04, 95%
CI [0.55, 1.53])
  - The effect of Depths [50-75] is statistically significant and positive (beta
= 32.48, 95% CI [12.14, 52.82], t(41) = 3.23, p = 0.002; Std. beta = 0.81, 95%
CI [0.30, 1.32])
  - The effect of Location [Reforma Waterfalls] is statistically non-significant
and positive (beta = 25.37, 95% CI [-17.73, 68.46], t(41) = 1.19, p = 0.241;
Std. beta = 0.64, 95% CI [-0.44, 1.72])
  - The effect of Location [Miguelito Dike] is statistically non-significant and
positive (beta = 11.02, 95% CI [-25.52, 47.55], t(41) = 0.61, p = 0.546; Std.
beta = 0.28, 95% CI [-0.64, 1.20])

Standardized parameters were obtained by fitting the model on a standardized
version of the dataset. 95% Confidence Intervals (CIs) and p-values were
computed using a Wald t-distribution approximation.</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb77"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Sig dif emmeans</span></span>
<span><span class="va">qf_lmer_means</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">qflmer</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Depths</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># letters</span></span>
<span><span class="va">cld_resultsqf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">qf_lmer_means</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert df</span></span>
<span><span class="va">emmeans_dfqf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">cld_resultsqf</span><span class="op">)</span></span>
<span><span class="va">emmeans_dfqf</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Depths emmean    SE   df lower.CL upper.CL .group
 0-15      202 10.01 21.7      181      222  a    
 16-30     228  9.91 21.2      207      248   b   
 50-75     234 10.50 24.8      212      256   b   
 31-45     243  9.90 21.1      222      264   b   

Results are averaged over the levels of: Location 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 4 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb79"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># plot</span></span>
<span><span class="va">qfplot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Depths</span>, y <span class="op">=</span> <span class="va">qf</span>, fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Depths</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">emmeans_dfqf</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">8</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"PD Faith"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>       caption <span class="op">=</span> <span class="st">"letters by lmer emmeans \n</span></span>
<span><span class="st">       taken depth, location and sediment core"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">depth_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></span>
<span><span class="va">qfplot</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-39-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section><section id="combine-plots" class="level3"><h3 class="anchored" data-anchor-id="combine-plots">Combine plots</h3>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>A combined plot of the observed richness and phylogenetic diversity was created because the former two indices showed significant differences.</p>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb80"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Combine plot</span></span>
<span></span>
<span><span class="va">ytitle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/ggdraw.html">ggdraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/draw_label.html">draw_label</a></span><span class="op">(</span><span class="st">"Effective number of ASVs"</span>, angle <span class="op">=</span> <span class="fl">90</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></span>
<span><span class="va">alphaplots</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">q0plot</span>, <span class="va">qfplot</span>, ncol <span class="op">=</span> <span class="fl">1</span>, rel_heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">alphaplots_y</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">ytitle</span>, <span class="va">alphaplots</span>, ncol <span class="op">=</span> <span class="fl">2</span>, rel_widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">1</span><span class="op">)</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>, <span class="st">""</span><span class="op">)</span><span class="op">)</span>    </span>
<span><span class="va">alphaplots_y</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-40-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode" id="cb81"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># save</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggsave.html">ggsave</a></span><span class="op">(</span><span class="st">"Figures/interior_mangroves/alpha_per_depth.pdf"</span>,<span class="va">alphaplots_y</span>, width <span class="op">=</span> <span class="fl">6</span>, height <span class="op">=</span> <span class="fl">8</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section></section><section id="by-location" class="level2"><h2 class="anchored" data-anchor-id="by-location">By location</h2>
<section id="q0-observed-richness-1" class="level3"><h3 class="anchored" data-anchor-id="q0-observed-richness-1">q0 Observed Richness</h3>
<div class="cell">
<div class="sourceCode" id="cb82"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmm</span></span>
<span><span class="va">q0lmerloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">q0</span> <span class="op">~</span> <span class="fu"><a href="https://rdrr.io/r/base/Log.html">log</a></span><span class="op">(</span><span class="va">Effective_reads</span><span class="op">)</span> <span class="op">+</span> <span class="va">Location</span> <span class="op">+</span> <span class="va">Depths</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co"># summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">q0lmerloc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: q0 ~ log(Effective_reads) + Location + Depths + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 680

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-2.3093 -0.5309 -0.0431  0.4842  1.8144 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 210009   458     
 Residual             209059   457     
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error       df t value Pr(&gt;|t|)    
(Intercept)                -5467.48    2792.66    37.52   -1.96  0.05772 .  
log(Effective_reads)         643.23     232.78    37.02    2.76  0.00886 ** 
LocationReforma Waterfalls   752.10     394.68     9.85    1.91  0.08628 .  
LocationMiguelito Dike       396.38     334.63     9.97    1.18  0.26366    
Depths16-30                  444.48     180.16    32.87    2.47  0.01900 *  
Depths31-45                  699.07     181.59    32.94    3.85  0.00052 ***
Depths50-75                  547.05     189.70    33.13    2.88  0.00686 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) lg(E_) LctnRW LctnMD D16-30 D31-45
lg(Effctv_) -0.995                                   
LctnRfrmWtr -0.017 -0.044                            
LoctnMgltDk -0.028 -0.044  0.517                     
Depths16-30  0.063 -0.095  0.004  0.004              
Depths31-45  0.124 -0.157  0.007  0.007  0.507       
Depths50-75  0.018 -0.045 -0.039 -0.046  0.475  0.474</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb84"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Permutation test </span></span>
<span><span class="va">perm_testloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lmerTest/man/ranova.html">rand</a></span><span class="op">(</span><span class="va">q0lmerloc</span>, nsim <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">perm_testloc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
q0 ~ log(Effective_reads) + Location + Depths + (1 | ID)
         npar logLik AIC  LRT Df Pr(&gt;Chisq)   
&lt;none&gt;      9   -340 698                      
(1 | ID)    8   -346 707 10.8  1      0.001 **
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode" id="cb86"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#effects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">q0lmerloc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Type III Analysis of Variance Table with Satterthwaite's method
                      Sum Sq Mean Sq NumDF DenDF F value Pr(&gt;F)   
log(Effective_reads) 1596301 1596301     1  37.0    7.64 0.0089 **
Location              770456  385228     2   9.8    1.84 0.2094   
Depths               3412939 1137646     3  33.0    5.44 0.0037 **
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb88"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># emmeans</span></span>
<span><span class="va">q0_lmer_means_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">q0lmerloc</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Location</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># significance letters</span></span>
<span><span class="va">q0cld_results_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q0_lmer_means_loc</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to data frame</span></span>
<span><span class="va">q0emmeans_df_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">q0cld_results_loc</span><span class="op">)</span></span>
<span><span class="va">q0emmeans_df_loc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Location           emmean  SE    df lower.CL upper.CL .group
 Fossil Lagoon        2739 262 10.58     2159     3318  a    
 Miguelito Dike       3135 209  9.72     2667     3603  a    
 Reforma Waterfalls   3491 296  9.69     2829     4152  a    

Results are averaged over the levels of: Depths 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 3 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb90"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># boxplot with significance</span></span>
<span><span class="va">q0plotloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Location</span>, y <span class="op">=</span> <span class="va">q0</span>, fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">q0emmeans_df_loc</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">5</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Observed Richness"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span> <span class="op">)</span> <span class="op">+</span> <span class="co">#,</span></span>
<span>       <span class="co">#caption = "letters by lmer emmeans") +</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">loc_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>, axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="q1-exp-shannon-diversity" class="level3"><h3 class="anchored" data-anchor-id="q1-exp-shannon-diversity">q1 Exp Shannon diversity</h3>
<div class="cell">
<div class="sourceCode" id="cb91"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmm</span></span>
<span><span class="va">q1lmerloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">q1</span> <span class="op">~</span> <span class="va">Location</span> <span class="op">+</span> <span class="va">Depths</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co"># summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">q1lmerloc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: q1 ~ Location + Depths + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 627

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-1.6697 -0.6187  0.0827  0.3962  2.4515 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 35546    189     
 Residual             47469    218     
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error    df t value Pr(&gt;|t|)    
(Intercept)                   596.2      121.9  15.3    4.89  0.00019 ***
LocationReforma Waterfalls     98.3      168.0  10.3    0.59  0.57086    
LocationMiguelito Dike         23.7      142.5  10.5    0.17  0.87137    
Depths16-30                   214.1       85.5  34.2    2.51  0.01716 *  
Depths31-45                   165.5       85.5  34.2    1.94  0.06116 .  
Depths50-75                   131.8       90.3  34.5    1.46  0.15340    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) LctnRW LctnMD D16-30 D31-45
LctnRfrmWtr -0.598                            
LoctnMgltDk -0.704  0.518                     
Depths16-30 -0.351  0.000  0.000              
Depths31-45 -0.351  0.000  0.000  0.500       
Depths50-75 -0.289 -0.045 -0.053  0.473  0.473</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb93"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Permutation test </span></span>
<span><span class="va">perm_testlocq1</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lmerTest/man/ranova.html">rand</a></span><span class="op">(</span><span class="va">q1lmerloc</span>, nsim <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">perm_testlocq1</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
q1 ~ Location + Depths + (1 | ID)
         npar logLik AIC  LRT Df Pr(&gt;Chisq)   
&lt;none&gt;      8   -313 643                      
(1 | ID)    7   -318 650 8.94  1     0.0028 **
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode" id="cb95"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#effects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">q1lmerloc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Type III Analysis of Variance Table with Satterthwaite's method
         Sum Sq Mean Sq NumDF DenDF F value Pr(&gt;F)  
Location  17497    8749     2  10.2    0.18  0.834  
Depths   327747  109249     3  34.3    2.30  0.095 .
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb97"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Differences emmeans</span></span>
<span><span class="va">q1lmer_means_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">q1lmerloc</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Location</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># significance letters</span></span>
<span><span class="va">q1cld_results_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q1lmer_means_loc</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to data frame</span></span>
<span><span class="va">q1emmeans_df_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">q1cld_results_loc</span><span class="op">)</span></span>
<span><span class="va">q1emmeans_df_loc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Location           emmean    SE    df lower.CL upper.CL .group
 Fossil Lagoon         724 111.5 10.65      478      970  a    
 Miguelito Dike        748  88.9  9.76      549      946  a    
 Reforma Waterfalls    822 125.7  9.76      541     1103  a    

Results are averaged over the levels of: Depths 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 3 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb99"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># boxplot with significance</span></span>
<span><span class="va">q1plotloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Location</span>, y <span class="op">=</span> <span class="va">q1</span>, fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">q1emmeans_df_loc</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">5</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Exp Shannon"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span> <span class="op">)</span> <span class="op">+</span> <span class="co">#,</span></span>
<span>       <span class="co">#caption = "letters by lmer emmeans") +</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">loc_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>, axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="q2-dominance-inv-simpson-1" class="level3"><h3 class="anchored" data-anchor-id="q2-dominance-inv-simpson-1">q2 Dominance Inv Simpson</h3>
<div class="cell">
<div class="sourceCode" id="cb100"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmm</span></span>
<span><span class="va">q2lmerloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">q2</span> <span class="op">~</span> <span class="va">Location</span> <span class="op">+</span> <span class="va">Depths</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co"># summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">q2lmerloc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: q2 ~ Location + Depths + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 542

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-1.7063 -0.6064  0.0468  0.5556  2.3862 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 2840     53.3    
 Residual             7832     88.5    
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error     df t value Pr(&gt;|t|)    
(Intercept)                  166.94      41.16  18.56    4.06   0.0007 ***
LocationReforma Waterfalls    10.23      53.74  10.38    0.19   0.8527    
LocationMiguelito Dike         1.72      45.70  10.62    0.04   0.9706    
Depths16-30                   87.60      34.71  34.19    2.52   0.0164 *  
Depths31-45                   31.89      34.71  34.19    0.92   0.3647    
Depths50-75                   35.58      36.61  34.70    0.97   0.3378    
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) LctnRW LctnMD D16-30 D31-45
LctnRfrmWtr -0.570                            
LoctnMgltDk -0.671  0.525                     
Depths16-30 -0.422  0.000  0.000              
Depths31-45 -0.422  0.000  0.000  0.500       
Depths50-75 -0.349 -0.056 -0.066  0.474  0.474</code></pre>
</div>
</div>
<pre class="{# Permutation test}"><code>perm_testlocq2 &lt;- rand(q2lmerloc, nsim = 1000)
perm_testlocq2

#effects
anova(q2lmerloc)</code></pre>
<div class="cell">
<div class="sourceCode" id="cb103"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Differences  emmeans</span></span>
<span><span class="va">q2lmer_means_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">q2lmerloc</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Location</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Significance </span></span>
<span><span class="va">q2cld_results_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">q2lmer_means_loc</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convertir a data frame</span></span>
<span><span class="va">q2emmeans_df_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">q2cld_results_loc</span><span class="op">)</span></span>
<span><span class="va">q2emmeans_df_loc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Location           emmean   SE    df lower.CL upper.CL .group
 Fossil Lagoon         206 35.9 10.99      127      285  a    
 Miguelito Dike        207 28.3  9.63      144      271  a    
 Reforma Waterfalls    216 40.0  9.63      126      306  a    

Results are averaged over the levels of: Depths 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 3 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb105"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># boxplot with significance</span></span>
<span><span class="va">q2plotloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Location</span>, y <span class="op">=</span> <span class="va">q2</span>, fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">q2emmeans_df_loc</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">5</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Exp Shannon"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span> <span class="op">)</span> <span class="op">+</span> <span class="co">#,</span></span>
<span>       <span class="co">#caption = "letters by lmer emmeans") +</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">loc_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>, axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="qf-faith-diversity-1" class="level3"><h3 class="anchored" data-anchor-id="qf-faith-diversity-1">qf Faith Diversity</h3>
<div class="cell">
<div class="sourceCode" id="cb106"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># lmm</span></span>
<span><span class="va">qflmerloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lme4/man/lmer.html">lmer</a></span><span class="op">(</span><span class="va">qf</span> <span class="op">~</span> <span class="va">Location</span> <span class="op">+</span> <span class="va">Depths</span> <span class="op">+</span> <span class="op">(</span><span class="fl">1</span> <span class="op">|</span> <span class="va">ID</span><span class="op">)</span>, data <span class="op">=</span> <span class="va">metadata_with_hill</span><span class="op">)</span></span>
<span><span class="co"># summary</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">qflmerloc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Linear mixed model fit by REML. t-tests use Satterthwaite's method [
lmerModLmerTest]
Formula: qf ~ Location + Depths + (1 | ID)
   Data: metadata_with_hill

REML criterion at convergence: 440

Scaled residuals: 
    Min      1Q  Median      3Q     Max 
-2.8154 -0.5311  0.0331  0.5800  1.6075 

Random effects:
 Groups   Name        Variance Std.Dev.
 ID       (Intercept) 796      28.2    
 Residual             621      24.9    
Number of obs: 50, groups:  ID, 13

Fixed effects:
                           Estimate Std. Error     df t value Pr(&gt;|t|)    
(Intercept)                  184.66      16.65  13.42   11.09  3.9e-08 ***
LocationReforma Waterfalls    27.53      23.72  10.23    1.16   0.2722    
LocationMiguelito Dike        12.86      20.10  10.33    0.64   0.5362    
Depths16-30                   28.22       9.77  34.14    2.89   0.0067 ** 
Depths31-45                   44.98       9.77  34.14    4.60  5.5e-05 ***
Depths50-75                   33.46      10.33  34.36    3.24   0.0027 ** 
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Correlation of Fixed Effects:
            (Intr) LctnRW LctnMD D16-30 D31-45
LctnRfrmWtr -0.615                            
LoctnMgltDk -0.726  0.515                     
Depths16-30 -0.293  0.000  0.000              
Depths31-45 -0.293  0.000  0.000  0.500       
Depths50-75 -0.241 -0.037 -0.044  0.473  0.473</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb108"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Permutation test </span></span>
<span><span class="va">perm_testlocqf</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/lmerTest/man/ranova.html">rand</a></span><span class="op">(</span><span class="va">qflmerloc</span>, nsim <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span>
<span><span class="va">perm_testlocqf</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>ANOVA-like table for random-effects: Single term deletions

Model:
qf ~ Location + Depths + (1 | ID)
         npar logLik AIC  LRT Df Pr(&gt;Chisq)    
&lt;none&gt;      8   -220 456                       
(1 | ID)    7   -228 470 15.9  1    6.6e-05 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
<div class="sourceCode" id="cb110"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#effects</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/anova.html">anova</a></span><span class="op">(</span><span class="va">qflmerloc</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>Type III Analysis of Variance Table with Satterthwaite's method
         Sum Sq Mean Sq NumDF DenDF F value  Pr(&gt;F)    
Location    838     419     2  10.2    0.67 0.53066    
Depths    14117    4706     3  34.2    7.58 0.00051 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1</code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb112"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co"># Differences emmeans</span></span>
<span><span class="va">qflmer_means_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rvlenth.github.io/emmeans/reference/emmeans.html">emmeans</a></span><span class="op">(</span><span class="va">qflmerloc</span>, <span class="va">pairwise</span> <span class="op">~</span> <span class="va">Location</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Significance</span></span>
<span><span class="va">qfcld_results_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/multcomp/man/cld.html">cld</a></span><span class="op">(</span>object <span class="op">=</span> <span class="va">qflmer_means_loc</span><span class="op">$</span><span class="va">emmeans</span>, Letters <span class="op">=</span> <span class="va">letters</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Convert to data frame</span></span>
<span><span class="va">qfemmeans_df_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">qfcld_results_loc</span><span class="op">)</span></span>
<span><span class="va">qfemmeans_df_loc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> Location           emmean   SE    df lower.CL upper.CL .group
 Fossil Lagoon         211 15.7 10.44      177      246  a    
 Miguelito Dike        224 12.6  9.84      196      252  a    
 Reforma Waterfalls    239 17.8  9.84      199      279  a    

Results are averaged over the levels of: Depths 
Degrees-of-freedom method: kenward-roger 
Confidence level used: 0.95 
P value adjustment: tukey method for comparing a family of 3 estimates 
significance level used: alpha = 0.05 
NOTE: If two or more means share the same grouping symbol,
      then we cannot show them to be different.
      But we also did not show them to be the same. </code></pre>
</div>
</div>
<div class="cell">
<div class="sourceCode" id="cb114"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="va">qfplotloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">metadata_with_hill</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Location</span>, y <span class="op">=</span> <span class="va">qf</span>, fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html">geom_jitter</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>fill <span class="op">=</span> <span class="va">Location</span><span class="op">)</span>, width <span class="op">=</span> <span class="fl">0.1</span>, alpha <span class="op">=</span> <span class="fl">0.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_text.html">geom_text</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">qfemmeans_df_loc</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>y <span class="op">=</span> <span class="va">emmean</span>, label <span class="op">=</span> <span class="va">.group</span><span class="op">)</span>, vjust <span class="op">=</span> <span class="op">-</span><span class="fl">5</span>, color <span class="op">=</span> <span class="st">"black"</span>, fontface <span class="op">=</span> <span class="st">"bold"</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_dodge.html">position_dodge</a></span><span class="op">(</span><span class="fl">0.9</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Exp Shannon"</span>, y <span class="op">=</span> <span class="cn">NULL</span>, x <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>       caption <span class="op">=</span> <span class="st">"letters by lmer emmeans"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_classic</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html">scale_fill_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="va">loc_colors</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span><span class="op">(</span>legend.position <span class="op">=</span> <span class="st">"none"</span>, axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_blank</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section><section id="combine-plots-1" class="level3"><h3 class="anchored" data-anchor-id="combine-plots-1">Combine plots</h3>
<div class="cell">
<div class="sourceCode" id="cb115"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="co">#Combine plot</span></span>
<span></span>
<span><span class="va">loctitle</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/ggdraw.html">ggdraw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/draw_label.html">draw_label</a></span><span class="op">(</span><span class="st">"Effective number of ASVs"</span>, angle <span class="op">=</span> <span class="fl">90</span>, size <span class="op">=</span> <span class="fl">12</span><span class="op">)</span></span>
<span><span class="va">alphaplotsloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">q0plotloc</span>, <span class="va">q1plotloc</span>,</span>
<span>                      <span class="va">q2plotloc</span>, <span class="va">qfplotloc</span>, </span>
<span>                      ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">alphaplots_loc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://wilkelab.org/cowplot/reference/plot_grid.html">plot_grid</a></span><span class="op">(</span><span class="va">ytitle</span>, <span class="va">alphaplotsloc</span>, ncol <span class="op">=</span> <span class="fl">2</span>, rel_widths <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.05</span>, <span class="fl">1</span><span class="op">)</span><span class="op">)</span>    </span>
<span></span>
<span><span class="va">alphaplots_loc</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure"><p><img src="03.2.AlphaDiversity_files/figure-html/unnamed-chunk-56-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
<div class="callout callout-style-default callout-important callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p>A comparison of alpha diversity indices revealed no significant differences among the mangrove sites. Consequently, incorporation of these into the final plot was unnecessary.</p>
</div>
</div>
</section></section></section><section id="save-rds-plots" class="level1"><h1>Save rds plots</h1>
<div class="cell">
<div class="sourceCode" id="cb116"><pre class="downlit sourceCode r code-with-copy"><code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/readRDS.html">saveRDS</a></span><span class="op">(</span><span class="va">alphaplots_y</span>, <span class="st">"rds/interior_mangroves/alpha_diversity_interior_depth_plot.rds"</span><span class="op">)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>


</section></main><!-- /main --><script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
    const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
    const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
    let newTheme = '';
    if(darkModeDefault) {
      newTheme = isAlternate ? baseTheme : alternateTheme;
    } else {
      newTheme = isAlternate ? alternateTheme : baseTheme;
    }
    const changeGiscusTheme = () => {
      // From: https://github.com/giscus/giscus/issues/336
      const sendMessage = (message) => {
        const iframe = document.querySelector('iframe.giscus-frame');
        if (!iframe) return;
        iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
      }
      sendMessage({
        setConfig: {
          theme: newTheme
        }
      });
    }
    const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
    if (isGiscussLoaded) {
      changeGiscusTheme();
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  const darkModeDefault = false;
  let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
    toggleGiscusIfUsed(toAlternate, darkModeDefault);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><nav class="page-navigation"><div class="nav-page nav-page-previous">
      <a href="./03.1.RelativeAbundances.html" class="pagination-link" aria-label="Relative Abundances">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Relative Abundances</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./03.3.BetaDiversity.html" class="pagination-link" aria-label="Beta diversity">
        <span class="nav-page-text">Beta diversity</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer"><div class="nav-footer">
    <div class="nav-footer-left">
<p>© Copyright 2025, Diana Hernández-Oaxaca</p>
</div>   
    <div class="nav-footer-center">
<p>LandaLab</p>
</div>
    <div class="nav-footer-right">
<p>This page is built with ❤ Science 🦠 🧬 and <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>


</body></html>